--- a/build/apple/tweak_info_plist.py
+++ b/build/apple/tweak_info_plist.py
@@ -116,7 +116,7 @@ def _AddVersionKeys(plist, version_forma
     VERSION_FILE = os.path.join(TOP, 'chrome/VERSION')
     (stdout, retval) = _GetOutput([
         VERSION_TOOL, '-f', VERSION_FILE, '-t',
-        '@MAJOR@.@MINOR@.@BUILD@.@PATCH@'
+        '@HELIUM_MAJOR@.@HELIUM_MINOR@.@HELIUM_PATCH@.@HELIUM_PLATFORM@'
     ])
 
     # If the command finished with a non-zero return code, then report the
@@ -423,7 +423,7 @@ def Main(argv):
         # http://lists.apple.com/archives/carbon-dev/2006/Jun/msg00139.html
         # BUILD will always be an increasing value, so BUILD_PATH gives us
         # something unique that meetings what LS wants.
-        'CFBundleVersion': '@BUILD@.@PATCH@',
+        'CFBundleVersion': '@MAJOR@.@MINOR@.@BUILD@.@PATCH@',
     }
   else:
     version_format_for_key = {
--- a/base/version_info/version_info_values.h.version
+++ b/base/version_info/version_info_values.h.version
@@ -7,6 +7,7 @@
 
 #define PRODUCT_NAME "@PRODUCT_FULLNAME@"
 #define PRODUCT_VERSION "@MAJOR@.@MINOR@.@BUILD@.@PATCH@"
+#define HELIUM_PRODUCT_VERSION "@HELIUM_MAJOR@.@HELIUM_MINOR@.@HELIUM_PATCH@.@HELIUM_PLATFORM@"
 #define LAST_CHANGE "@LASTCHANGE@"
 #define IS_OFFICIAL_BUILD @OFFICIAL_BUILD@
 
--- a/base/version_info/version_info.h
+++ b/base/version_info/version_info.h
@@ -25,6 +25,10 @@ constexpr std::string_view GetProductNam
   return PRODUCT_NAME;
 }
 
+constexpr std::string_view GetHeliumVersionNumber() {
+  return HELIUM_PRODUCT_VERSION;
+}
+
 // Returns the version number, e.g. "6.0.490.1".
 constexpr std::string_view GetVersionNumber() {
   return PRODUCT_VERSION;
--- a/chrome/browser/ui/webui/version/version_ui.cc
+++ b/chrome/browser/ui/webui/version/version_ui.cc
@@ -235,6 +235,8 @@ void VersionUI::AddVersionDetailStrings(
                                   VersionProcessorVariation());
 
   // Data strings.
+  html_source->AddString(version_ui::kHeliumVersion,
+                         version_info::GetHeliumVersionNumber());
   html_source->AddString(version_ui::kVersion,
                          version_info::GetVersionNumber());
   html_source->AddString(version_ui::kVersionSuffix,
@@ -328,12 +330,14 @@ void VersionUI::AddVersionDetailStrings(
 std::u16string VersionUI::GetAnnotatedVersionStringForUi() {
   return l10n_util::GetStringFUTF16(
       IDS_SETTINGS_ABOUT_PAGE_BROWSER_VERSION,
+      {base::UTF8ToUTF16(version_info::GetHeliumVersionNumber()),
       base::UTF8ToUTF16(version_info::GetVersionNumber()),
       base::UTF8ToUTF16(GetVersionInformationalSuffix()),
       l10n_util::GetStringUTF16(version_info::IsOfficialBuild()
                                     ? IDS_VERSION_UI_OFFICIAL
                                     : IDS_VERSION_UI_UNOFFICIAL),
       base::UTF8ToUTF16(GetProductModifier()),
-      l10n_util::GetStringUTF16(VersionUI::VersionProcessorVariation()));
+      l10n_util::GetStringUTF16(VersionUI::VersionProcessorVariation()),
+      u"Chromium"}, nullptr);
 }
 #endif  // !BUILDFLAG(IS_ANDROID)
--- a/components/webui/version/version_ui_constants.cc
+++ b/components/webui/version/version_ui_constants.cc
@@ -88,6 +88,7 @@ const char kVariationsSourceName[] = "va
 const char kVariationsSeed[] = "variations_seed";
 const char kVariationsSeedName[] = "variations_seed_name";
 const char kVersion[] = "version";
+const char kHeliumVersion[] = "helium_version";
 const char kVersionSuffix[] = "version_suffix";
 const char kVersionModifier[] = "version_modifier";
 const char kVersionProcessorVariation[] = "version_processor_variation";
--- a/components/webui/version/version_ui_constants.h
+++ b/components/webui/version/version_ui_constants.h
@@ -92,6 +92,7 @@ extern const char kVariationsSourceName[
 extern const char kVariationsSeed[];
 extern const char kVariationsSeedName[];
 extern const char kVersion[];
+extern const char kHeliumVersion[];
 extern const char kVersionSuffix[];
 extern const char kVersionModifier[];
 extern const char kVersionProcessorVariation[];
--- a/chrome/app/settings_strings.grdp
+++ b/chrome/app/settings_strings.grdp
@@ -42,7 +42,7 @@
 
   <!-- About Page -->
   <message name="IDS_SETTINGS_ABOUT_PAGE_BROWSER_VERSION" desc="The text label describing the version of the browser, example: Version 57.0.2937.0-r123456 (Developer Build) unknown (64-bit). The suffix of the version (eg. '-r123456') exists only in limited cases.)">
-    Version <ph name="PRODUCT_VERSION">$1<ex>15.0.865.0</ex></ph><ph name="PRODUCT_VERSION_SUFFIX">$2<ex>-r123456</ex></ph>  (<ph name="PRODUCT_CHANNEL">$3<ex>Developer Build</ex></ph>) <ph name="PRODUCT_MODIFIER">$4</ph> <ph name="PRODUCT_VERSION_BITS">$5</ph>
+    Version <ph name="HELIUM_PRODUCT_VERSION">$1<ex>0.0.0.0</ex></ph> (<ph name="PRODUCT_CHANNEL">$4<ex>Developer Build</ex></ph>, <ph name="CHROMIUM_NAME">$7</ph> <ph name="PRODUCT_VERSION">$2<ex>15.0.865.0</ex></ph><ph name="PRODUCT_VERSION_SUFFIX">$3<ex>-r123456</ex></ph>) <ph name="PRODUCT_MODIFIER">$5</ph> <ph name="PRODUCT_VERSION_BITS">$6</ph>
   </message>
   <message name="IDS_SETTINGS_ABOUT_PAGE_LEARN_MORE_UPDATE_ERRORS" desc="The accessibility label for a 'Learn more' link next to an error message about how an update has failed.">
     Learn more about fixing update errors
--- a/components/webui/version/resources/about_version.html
+++ b/components/webui/version/resources/about_version.html
@@ -54,11 +54,21 @@ about:version template page
       <table id="inner" cellpadding="0" cellspacing="0" border="0">
         <tr><td class="label">$i18n{application_label}</td>
           <td class="version" id="version">
+            <span id="copy-content-helium">
+              <span>$i18n{helium_version}</span>
+              (<span>$i18n{official}</span>)
+              <span>$i18n{version_processor_variation}</span>
+            </span>
+            <button id="copy-to-clipboard-helium" aria-label="$i18n{copy_label}">
+              <div id="copy-to-clipboard-icon"></div>
+            </button>
+          </td>
+        </tr>
+        <tr><td class="label">Chromium</td>
+          <td class="version" id="version">
             <span id="copy-content">
               <span>$i18n{version}$i18n{version_suffix}</span>
-              (<span>$i18n{official}</span>)
               <span>$i18n{version_modifier}</span>
-              <span>$i18n{version_processor_variation}</span>
   <if expr="is_win">
               <span>$i18n{update_cohort_name}</span>
   </if>
--- a/components/webui/version/resources/about_version.ts
+++ b/components/webui/version/resources/about_version.ts
@@ -134,6 +134,12 @@ async function copyVersionToClipboard()
   announceCopy('copy_notice');
 }
 
+async function copyHeliumVersionToClipboard() {
+  await navigator.clipboard.writeText(
+      getRequiredElement('copy-content-helium').innerText);
+  announceCopy('copy_notice');
+}
+
 async function copyVariationsToClipboard() {
   const cmdLine =
       getRequiredElement('variations-cmd').dataset['value'] as string;
@@ -202,6 +208,9 @@ function initialize() {
   getRequiredElement('copy-to-clipboard')
       .addEventListener('click', copyVersionToClipboard);
 
+  getRequiredElement('copy-to-clipboard-helium')
+      .addEventListener('click', copyHeliumVersionToClipboard);
+
   getRequiredElement('copy-variations-to-clipboard')
       .addEventListener('click', copyVariationsToClipboard);
 }
--- a/components/webui/version/resources/about_version.css
+++ b/components/webui/version/resources/about_version.css
@@ -90,6 +90,7 @@ body {
   vertical-align: bottom;
 }
 
+#copy-to-clipboard-helium,
 #copy-to-clipboard,
 #copy-variations-to-clipboard {
   background-color: var(--background-color);
--- a/chrome/app/chrome_main_delegate.cc
+++ b/chrome/app/chrome_main_delegate.cc
@@ -367,16 +367,16 @@ bool HandleCreditsSwitch(const base::Com
 bool HandleVersionSwitches(const base::CommandLine& command_line) {
 #if !BUILDFLAG(IS_MAC)
   if (command_line.HasSwitch(switches::kProductVersion)) {
-    UNSAFE_TODO(printf("%s\n", version_info::GetVersionNumber().data()));
+    UNSAFE_TODO(printf("%s\n", version_info::GetHeliumVersionNumber().data()));
     return true;
   }
 #endif
 
   if (command_line.HasSwitch(switches::kVersion)) {
     UNSAFE_TODO(printf(
-        "%s %s %s\n", version_info::GetProductName().data(),
-        version_info::GetVersionNumber().data(),
-        chrome::GetChannelName(chrome::WithExtendedStable(true)).c_str()));
+        "%s %s (Chromium %s)\n", version_info::GetProductName().data(),
+        version_info::GetHeliumVersionNumber().data(),
+        version_info::GetVersionNumber().data()));
     return true;
   }
 
