--- a/chrome/browser/ui/views/frame/system_menu_model_builder.cc
+++ b/chrome/browser/ui/views/frame/system_menu_model_builder.cc
@@ -104,6 +104,12 @@ void SystemMenuModelBuilder::BuildSystem
   model->AddItemWithStringId(IDC_BOOKMARK_ALL_TABS, IDS_BOOKMARK_ALL_TABS);
   model->AddItemWithStringId(IDC_NAME_WINDOW, IDS_NAME_WINDOW);
 
+  if (!browser()->profile()->IsOffTheRecord()) {
+    model->AddSeparator(ui::NORMAL_SEPARATOR);
+    model->AddItemWithStringId(IDC_SHOW_CUSTOMIZE_CHROME_SIDE_PANEL,
+                               IDS_SHOW_CUSTOMIZE_CHROME_SIDE_PANEL);
+  }
+
   if (base::FeatureList::IsEnabled(tabs::kHorizontalTabStripComboButton)) {
     model->AddSeparator(ui::NORMAL_SEPARATOR);
     model->AddItemWithStringId(IDC_TAB_SEARCH_TOGGLE_PIN,
