--- a/chrome/browser/renderer_context_menu/render_view_context_menu.cc
+++ b/chrome/browser/renderer_context_menu/render_view_context_menu.cc
@@ -1194,11 +1194,6 @@ void RenderViewContextMenu::InitMenu() {
     AppendSpellingAndSearchSuggestionItems();
   }
 
-  if (content_type_->SupportsGroup(
-          ContextMenuContentType::ITEM_GROUP_AUTOFILL)) {
-    autofill_context_menu_manager_.AppendItems();
-  }
-
   if (editable) {
     AppendOtherEditableItems();
   }
@@ -2303,7 +2298,6 @@ void RenderViewContextMenu::AppendPageIt
                                   IDS_CONTENT_CONTEXT_SAVEPAGEAS);
   menu_model_.AddItemWithStringId(IDC_PRINT, IDS_CONTENT_CONTEXT_PRINT);
   AppendLiveCaptionItem();
-  AppendMediaRouterItem();
 
   if (features::IsReadAnythingMenuShuffleExperimentEnabled()) {
     // This will be set to false in AppendRegionSearchItem() if it is called.
@@ -2455,8 +2449,7 @@ void RenderViewContextMenu::AppendMediaR
 
 void RenderViewContextMenu::AppendReadAnythingItem() {
   // Show Read Anything option if it's not already open in the side panel.
-  if (GetBrowser() && GetBrowser()->is_type_normal() &&
-      !IsReadAnythingEntryShowing(GetBrowser())) {
+  if ((false)) {
     menu_model_.AddItemWithStringId(IDC_CONTENT_CONTEXT_OPEN_IN_READING_MODE,
                                     IDS_CONTENT_CONTEXT_READING_MODE);
   }
@@ -3728,30 +3721,9 @@ void RenderViewContextMenu::ExecuteComma
   }
 }
 
-void RenderViewContextMenu::AddSpellCheckServiceItem(bool is_checked) {
-  AddSpellCheckServiceItem(&menu_model_, is_checked);
-}
+void RenderViewContextMenu::AddSpellCheckServiceItem(bool) {}
 
-void RenderViewContextMenu::AddAccessibilityLabelsServiceItem(bool is_checked) {
-  if (is_checked) {
-    menu_model_.AddCheckItemWithStringId(
-        IDC_CONTENT_CONTEXT_ACCESSIBILITY_LABELS_TOGGLE,
-        IDS_CONTENT_CONTEXT_ACCESSIBILITY_LABELS_MENU_OPTION);
-  } else {
-    // Add the submenu if the whole feature is not enabled.
-    accessibility_labels_submenu_model_.AddItemWithStringId(
-        IDC_CONTENT_CONTEXT_ACCESSIBILITY_LABELS_TOGGLE,
-        IDS_CONTENT_CONTEXT_ACCESSIBILITY_LABELS_SEND);
-    accessibility_labels_submenu_model_.AddItemWithStringId(
-        IDC_CONTENT_CONTEXT_ACCESSIBILITY_LABELS_TOGGLE_ONCE,
-        IDS_CONTENT_CONTEXT_ACCESSIBILITY_LABELS_SEND_ONCE);
-    menu_model_.AddSubMenu(
-        IDC_CONTENT_CONTEXT_ACCESSIBILITY_LABELS,
-        l10n_util::GetStringUTF16(
-            IDS_CONTENT_CONTEXT_ACCESSIBILITY_LABELS_MENU_OPTION),
-        &accessibility_labels_submenu_model_);
-  }
-}
+void RenderViewContextMenu::AddAccessibilityLabelsServiceItem(bool) {}
 
 // static
 void RenderViewContextMenu::RegisterMenuShownCallbackForTesting(
@@ -4115,7 +4087,7 @@ bool RenderViewContextMenu::IsVideoFrame
 bool RenderViewContextMenu::AppendQRCodeGeneratorItem(bool for_image,
                                                       bool draw_icon,
                                                       bool add_separator) {
-  if (!IsQRCodeGeneratorEnabled()) {
+  if ((true)) {
     return false;
   }
 
--- a/chrome/browser/ui/browser_tab_strip_model_delegate.cc
+++ b/chrome/browser/ui/browser_tab_strip_model_delegate.cc
@@ -301,7 +301,7 @@ void BrowserTabStripModelDelegate::AddTo
 }
 
 bool BrowserTabStripModelDelegate::SupportsReadLater() {
-  return !browser_->profile()->IsGuestSession() && !IsForWebApp();
+  return false;
 }
 
 bool BrowserTabStripModelDelegate::IsForWebApp() {
