--- a/chrome/browser/ui/ui_features.cc
+++ b/chrome/browser/ui/ui_features.cc
@@ -192,14 +192,9 @@ BASE_FEATURE(kThreeButtonPasswordSaveDia
 
 BASE_FEATURE(kToolbarHeightSidePanel, base::FEATURE_DISABLED_BY_DEFAULT);
 
-BASE_FEATURE(kSidePanelFlyoverAnimation, base::FEATURE_ENABLED_BY_DEFAULT);
+BASE_FEATURE(kSidePanelFlyoverAnimation, base::FEATURE_DISABLED_BY_DEFAULT);
 bool UseSidePanelFlyoverAnimation() {
-#if BUILDFLAG(IS_MAC)
-  // Mac can smoothly resize contents and does not need flyover.
   return false;
-#else
-  return base::FeatureList::IsEnabled(kSidePanelFlyoverAnimation);
-#endif
 }
 
 // Enables enterprise profile badging for managed profiles on the toolbar avatar
