--- a/services/network/public/cpp/features.cc
+++ b/services/network/public/cpp/features.cc
@@ -441,7 +441,7 @@ BASE_FEATURE(kUpdateRequestForCorsRedire
 
 // https://github.com/patcg-individual-drafts/topics
 // Kill switch for the Topics API.
-BASE_FEATURE(kBrowsingTopics, base::FEATURE_ENABLED_BY_DEFAULT);
+BASE_FEATURE(kBrowsingTopics, base::FEATURE_DISABLED_BY_DEFAULT);
 
 // Enable the shared storage API. Note that enabling this feature does not
 // automatically expose this API to the web, it only allows the element to be
@@ -520,7 +520,7 @@ BASE_FEATURE(kSharedStorageTransactional
 
 // Kill switch for the Interest Group API, i.e. if disabled, the
 // API exposure will be disabled regardless of the OT config.
-BASE_FEATURE(kInterestGroupStorage, base::FEATURE_ENABLED_BY_DEFAULT);
+BASE_FEATURE(kInterestGroupStorage, base::FEATURE_DISABLED_BY_DEFAULT);
 // TODO(crbug.com/40176812): Adjust these limits in response to usage.
 BASE_FEATURE_PARAM(int,
                    kInterestGroupStorageMaxOwners,
