--- a/chrome/browser/ui/views/webid/fedcm_account_selection_view_desktop.cc
+++ b/chrome/browser/ui/views/webid/fedcm_account_selection_view_desktop.cc
@@ -1439,24 +1439,8 @@ void FedCmAccountSelectionView::ShouldSh
   }
 
   if (dialog_type_ == DialogType::BUBBLE) {
-    // Hide the bubble dialog if it can't fit.
-    if (!CanFitInWebContents()) {
-      should_show = false;
-    }
-
-    // Or if a popup is showing.
-    if (popup_window_) {
-      should_show = false;
-    }
-
-    // Or if we want to hide until Show*() is called.
-    if (hide_dialog_widget_after_idp_login_popup_) {
-      should_show = false;
-    }
-
-    if (!tab_->CanShowModalUI()) {
-      should_show = false;
-    }
+    should_show = false;
+    return;
   }
 }
 
