--- a/chrome/browser/about_flags.cc
+++ b/chrome/browser/about_flags.cc
@@ -414,11 +414,11 @@ namespace about_flags {
 
 namespace {
 
-const unsigned kOsAll = kOsMac | kOsWin | kOsLinux | kOsCrOS | kOsAndroid;
-const unsigned kOsDesktop = kOsMac | kOsWin | kOsLinux | kOsCrOS;
+const unsigned kOsAll = kOsMac | kOsWin | kOsLinux;
+const unsigned kOsDesktop = kOsMac | kOsWin | kOsLinux;
 
 #if defined(USE_AURA)
-const unsigned kOsAura = kOsWin | kOsLinux | kOsCrOS;
+const unsigned kOsAura = kOsWin | kOsLinux;
 #endif  // USE_AURA
 
 #if defined(USE_AURA)
@@ -4651,11 +4651,13 @@ const FeatureEntry::FeatureVariation
 // calculate and verify checksum.
 //
 // When adding a new choice, add it to the end of the list.
+#include "chrome/browser/helium_flag_choices.h"
 #include "chrome/browser/ungoogled_flag_choices.h"
 #include "chrome/browser/bromite_flag_choices.h"
 #include "chrome/browser/ungoogled_platform_flag_choices.h"
 #include "chrome/browser/existing_switch_flag_choices.h"
 const FeatureEntry kFeatureEntries[] = {
+#include "chrome/browser/helium_flag_entries.h"
 #include "chrome/browser/ungoogled_flag_entries.h"
 #include "chrome/browser/bromite_flag_entries.h"
 #include "chrome/browser/ungoogled_platform_flag_entries.h"
--- /dev/null
+++ b/chrome/browser/helium_flag_choices.h
@@ -0,0 +1,16 @@
+// Copyright 2025 The Helium Authors
+// You can use, redistribute, and/or modify this source code under
+// the terms of the GPL-3.0 license that can be found in the LICENSE file.
+
+#ifndef CHROME_BROWSER_HELIUM_FLAG_CHOICES_H_
+#define CHROME_BROWSER_HELIUM_FLAG_CHOICES_H_
+
+#include "components/webui/flags/feature_entry.h"
+#include "chrome/browser/channel_selection.h"
+
+namespace helium {
+  using namespace ::helium;
+
+}  // namespace helium
+
+#endif  /* CHROME_BROWSER_HELIUM_FLAG_CHOICES_H_ */
--- /dev/null
+++ b/chrome/browser/helium_flag_entries.h
@@ -0,0 +1,7 @@
+// Copyright 2025 The Helium Authors
+// You can use, redistribute, and/or modify this source code under
+// the terms of the GPL-3.0 license that can be found in the LICENSE file.
+
+#ifndef CHROME_BROWSER_HELIUM_FLAG_ENTRIES_H_
+#define CHROME_BROWSER_HELIUM_FLAG_ENTRIES_H_
+#endif  /* CHROME_BROWSER_HELIUM_FLAG_ENTRIES_H_ */
