--- a/content/public/common/url_constants.h
+++ b/content/public/common/url_constants.h
@@ -19,6 +19,7 @@ namespace content {
 inline constexpr char kChromeDevToolsScheme[] = "devtools";
 inline constexpr char kChromeErrorScheme[] = "chrome-error";
 inline constexpr char kChromeUIScheme[] = "chrome";  // Used for WebUIs.
+inline constexpr char kHeliumUIScheme[] = "helium";  // also used for webuis :3
 inline constexpr char kChromeNativeScheme[] = "chrome-native";
 inline constexpr char kChromeUIUntrustedScheme[] = "chrome-untrusted";
 inline constexpr char kViewSourceScheme[] = "view-source";
--- a/chrome/browser/profiles/profile_io_data.cc
+++ b/chrome/browser/profiles/profile_io_data.cc
@@ -44,6 +44,7 @@ bool ProfileIOData::IsHandledProtocol(st
       extensions::kExtensionScheme,
 #endif
       content::kChromeUIScheme,
+      content::kHeliumUIScheme,
       content::kChromeUIUntrustedScheme,
       url::kDataScheme,
 #if BUILDFLAG(IS_CHROMEOS)
--- a/content/common/url_schemes.cc
+++ b/content/common/url_schemes.cc
@@ -30,6 +30,7 @@ const char* const kDefaultSavableSchemes
   url::kFileSystemScheme,
   kChromeDevToolsScheme,
   kChromeUIScheme,
+  kHeliumUIScheme,
   url::kDataScheme
 };
 
@@ -60,6 +61,7 @@ void RegisterContentSchemes(bool should_
 
   url::AddStandardScheme(kChromeDevToolsScheme, url::SCHEME_WITH_HOST);
   url::AddStandardScheme(kChromeUIScheme, url::SCHEME_WITH_HOST);
+  url::AddStandardScheme(kHeliumUIScheme, url::SCHEME_WITH_HOST);
   url::AddStandardScheme(kChromeUIUntrustedScheme, url::SCHEME_WITH_HOST);
   url::AddStandardScheme(kChromeErrorScheme, url::SCHEME_WITH_HOST);
   for (auto& scheme : schemes.standard_schemes)
@@ -70,6 +72,7 @@ void RegisterContentSchemes(bool should_
 
   schemes.secure_schemes.push_back(kChromeDevToolsScheme);
   schemes.secure_schemes.push_back(kChromeUIScheme);
+  schemes.secure_schemes.push_back(kHeliumUIScheme);
   schemes.secure_schemes.push_back(kChromeUIUntrustedScheme);
   schemes.secure_schemes.push_back(kChromeErrorScheme);
   for (auto& scheme : schemes.secure_schemes)
@@ -90,6 +93,7 @@ void RegisterContentSchemes(bool should_
     url::AddNoAccessScheme(scheme.c_str());
 
   schemes.cors_enabled_schemes.push_back(kChromeUIScheme);
+  schemes.cors_enabled_schemes.push_back(kHeliumUIScheme);
   schemes.cors_enabled_schemes.push_back(kChromeUIUntrustedScheme);
   for (auto& scheme : schemes.cors_enabled_schemes)
     url::AddCorsEnabledScheme(scheme.c_str());
@@ -118,6 +122,7 @@ void RegisterContentSchemes(bool should_
   // service workers don't work for chrome[-untrusted]:// when the flag is not
   // enabled.
   schemes.service_worker_schemes.push_back(kChromeUIScheme);
+  schemes.service_worker_schemes.push_back(kHeliumUIScheme);
   schemes.service_worker_schemes.push_back(kChromeUIUntrustedScheme);
 
   // Prevent future modification of the scheme lists. This is to prevent
--- a/content/renderer/render_thread_impl.cc
+++ b/content/renderer/render_thread_impl.cc
@@ -862,6 +862,13 @@ void RenderThreadImpl::RegisterSchemes()
       chrome_scheme);
   WebSecurityPolicy::RegisterURLSchemeAsWebUI(chrome_scheme);
 
+  // helium:
+  WebString helium_scheme(WebString::FromAscii(kHeliumUIScheme));
+  WebSecurityPolicy::RegisterURLSchemeAsDisplayIsolated(helium_scheme);
+  WebSecurityPolicy::RegisterURLSchemeAsNotAllowingJavascriptURLs(
+      helium_scheme);
+  WebSecurityPolicy::RegisterURLSchemeAsWebUI(helium_scheme);
+
   WebString chrome_untrusted_scheme(
       WebString::FromAscii(kChromeUIUntrustedScheme));
 
--- a/chrome/browser/history/history_utils.cc
+++ b/chrome/browser/history/history_utils.cc
@@ -20,6 +20,7 @@ bool CanAddURLToHistory(const GURL& url)
       url.SchemeIs(url::kContentScheme) ||
       url.SchemeIs(content::kChromeDevToolsScheme) ||
       url.SchemeIs(content::kChromeUIScheme) ||
+      url.SchemeIs(content::kHeliumUIScheme) ||
       url.SchemeIs(content::kChromeUIUntrustedScheme) ||
       url.SchemeIs(content::kViewSourceScheme) ||
       url.SchemeIs(url::kTraceScheme) ||
--- a/chrome/browser/ui/incognito_allowed_url.cc
+++ b/chrome/browser/ui/incognito_allowed_url.cc
@@ -23,7 +23,8 @@ namespace {
 bool IsHostAllowedInIncognito(const GURL& url) {
   std::string scheme = url.GetScheme();
   std::string_view host = url.host();
-  if (scheme != content::kChromeUIScheme) {
+  if (scheme != content::kChromeUIScheme &&
+      scheme != content::kHeliumUIScheme) {
     return true;
   }
 
--- a/chrome/browser/ui/browser_navigator.cc
+++ b/chrome/browser/ui/browser_navigator.cc
@@ -573,6 +573,12 @@ base::WeakPtr<content::NavigationHandle>
     }
   }
 
+  if (params->url.scheme() == content::kHeliumUIScheme) {
+    GURL::Replacements replacements;
+    replacements.SetSchemeStr(content::kChromeUIScheme);
+    params->url = params->url.ReplaceComponents(replacements);
+  }
+
   // Open System Apps in their standalone window if necessary.
   // TODO(crbug.com/40136163): Remove this code after we integrate with intent
   // handling.
--- a/chrome/browser/extensions/extension_tab_util.cc
+++ b/chrome/browser/extensions/extension_tab_util.cc
@@ -1042,7 +1042,8 @@ bool ExtensionTabUtil::IsKillURL(const G
     return true;
   }
 
-  if (!url.SchemeIs(content::kChromeUIScheme)) {
+  if (!url.SchemeIs(content::kChromeUIScheme)
+      && !url.SchemeIs(content::kHeliumUIScheme)) {
     return false;
   }
 
--- a/chrome/browser/ui/browser.cc
+++ b/chrome/browser/ui/browser.cc
@@ -2199,6 +2199,13 @@ void Browser::SetContentsBounds(WebConte
 }
 
 void Browser::UpdateTargetURL(WebContents* source, const GURL& url) {
+  GURL patched_url = url;
+  if (patched_url.SchemeIs(content::kChromeUIScheme)) {
+    GURL::Replacements replacements;
+    replacements.SetSchemeStr(content::kHeliumUIScheme);
+    patched_url = patched_url.ReplaceComponents(replacements);
+  }
+
   std::vector<StatusBubble*> status_bubbles = GetStatusBubbles();
   for (StatusBubble* status_bubble : status_bubbles) {
     StatusBubbleViews* status_bubble_views =
@@ -2206,7 +2213,7 @@ void Browser::UpdateTargetURL(WebContent
     ContentsWebView* anchor =
         static_cast<ContentsWebView*>(status_bubble_views->base_view());
     if (source == anchor->GetWebContents()) {
-      status_bubble->SetURL(url);
+      status_bubble->SetURL(patched_url);
       break;
     }
   }
--- a/components/omnibox/browser/omnibox_text_util.cc
+++ b/components/omnibox/browser/omnibox_text_util.cc
@@ -161,6 +161,12 @@ void AdjustTextForCopy(
     *url_from_text = navigation_entry_url;
     *write_url = true;
 
+    if (url_from_text->scheme() == content::kChromeUIScheme) {
+      GURL::Replacements replacements;
+      replacements.SetSchemeStr(content::kHeliumUIScheme);
+      *url_from_text = url_from_text->ReplaceComponents(replacements);
+    }
+
     // Don't let users copy Reader Mode page URLs.
     // We display the original article's URL in the omnibox, so users will
     // expect that to be what is copied to the clipboard.
--- a/components/url_formatter/url_formatter.cc
+++ b/components/url_formatter/url_formatter.cc
@@ -20,6 +20,7 @@
 #include "base/strings/utf_string_conversions.h"
 #include "base/threading/thread_local_storage.h"
 #include "build/build_config.h"
+#include "content/public/common/url_constants.h"
 #include "components/url_formatter/spoof_checks/idn_spoof_checker.h"
 #include "net/base/registry_controlled_domains/registry_controlled_domain.h"
 #include "third_party/icu/source/common/unicode/uidna.h"
@@ -561,6 +562,7 @@ const FormatUrlType kFormatUrlTrimAfterH
 const FormatUrlType kFormatUrlOmitFileScheme = 1 << 7;
 const FormatUrlType kFormatUrlOmitMailToScheme = 1 << 8;
 const FormatUrlType kFormatUrlOmitMobilePrefix = 1 << 9;
+const FormatUrlType kFormatReplaceChromeProtocol = 1 << 10;
 
 const FormatUrlType kFormatUrlOmitDefaults =
     kFormatUrlOmitUsernamePassword | kFormatUrlOmitHTTP |
@@ -623,6 +625,18 @@ std::u16string FormatUrlWithAdjustments(
                                prefix_end, adjustments);
   }
 
+  if ((format_types & kFormatReplaceChromeProtocol) != 0) {
+    if (url.SchemeIs(content::kChromeUIScheme)) {
+      GURL url_;
+      GURL::Replacements replacements;
+      replacements.SetSchemeStr(content::kHeliumUIScheme);
+      url_ = url.ReplaceComponents(replacements);
+
+      return FormatUrlWithAdjustments(url_, format_types, unescape_rules,
+                                      new_parsed, prefix_end, adjustments);
+    }
+  }
+
   // We handle both valid and invalid URLs (this will give us the spec
   // regardless of validity).
   const std::string& spec = url.possibly_invalid_spec();
--- a/components/url_formatter/url_formatter.h
+++ b/components/url_formatter/url_formatter.h
@@ -101,6 +101,8 @@ extern const FormatUrlType kFormatUrlOmi
 // removal, or experimental flags.
 extern const FormatUrlType kFormatUrlOmitDefaults;
 
+// Replace chrome:// with our own protocol.
+extern const FormatUrlType kFormatReplaceChromeProtocol;
 // Creates a string representation of |url|. The IDN host name is turned to
 // Unicode if the Unicode representation is deemed safe. |format_type| is a
 // bitmask of FormatUrlTypes, see it for details. |unescape_rules| defines how
--- a/chrome/browser/ui/web_applications/app_browser_controller.cc
+++ b/chrome/browser/ui/web_applications/app_browser_controller.cc
@@ -251,7 +251,8 @@ std::u16string AppBrowserController::For
     url_formatter::FormatUrlTypes format_types) {
   auto origin = url::Origin::Create(url);
   return url_formatter::FormatUrl(origin.opaque() ? url : origin.GetURL(),
-                                  format_types, base::UnescapeRule::SPACES,
+                                  format_types | url_formatter::kFormatReplaceChromeProtocol,
+                                  base::UnescapeRule::SPACES,
                                   nullptr, nullptr, nullptr);
 }
 
--- a/components/omnibox/browser/location_bar_model_impl.cc
+++ b/components/omnibox/browser/location_bar_model_impl.cc
@@ -152,7 +152,8 @@ std::u16string LocationBarModelImpl::Get
   // the space.
   const std::u16string formatted_text =
       delegate_->FormattedStringWithEquivalentMeaning(
-          url, url_formatter::FormatUrl(url, format_types,
+          url, url_formatter::FormatUrl(url,
+                                        format_types | url_formatter::kFormatReplaceChromeProtocol,
                                         base::UnescapeRule::NORMAL, nullptr,
                                         nullptr, nullptr));
 
--- a/components/url_formatter/elide_url.cc
+++ b/components/url_formatter/elide_url.cc
@@ -426,7 +426,8 @@ std::u16string FormatUrlForDisplayOmitSc
       url_formatter::kFormatUrlOmitDefaults |
           url_formatter::kFormatUrlTrimAfterHost |
           url_formatter::kFormatUrlOmitHTTPS |
-          url_formatter::kFormatUrlOmitTrivialSubdomains,
+          url_formatter::kFormatUrlOmitTrivialSubdomains |
+          url_formatter::kFormatReplaceChromeProtocol,
       base::UnescapeRule::SPACES, nullptr, nullptr, nullptr);
 }
 
--- a/components/url_formatter/url_fixer.cc
+++ b/components/url_formatter/url_fixer.cc
@@ -39,6 +39,7 @@ namespace {
 
 // Hardcode these constants to avoid dependences on //chrome and //content.
 const char kChromeUIScheme[] = "chrome";
+const char kHeliumUIScheme[] = "helium";
 const char kDevToolsScheme[] = "devtools";
 const char kChromeUIDefaultHost[] = "version";
 const char kViewSourceScheme[] = "view-source";
@@ -446,7 +447,8 @@ std::string SegmentURLInternal(std::stri
       (*text)[semicolon] = ':';
       if (GetValidScheme(*text, &parts->scheme, &scheme) &&
           (url::IsStandard(scheme) || scheme == url::kAboutScheme ||
-           scheme == kChromeUIScheme || scheme == url::kFileScheme)) {
+           scheme == kChromeUIScheme || scheme == kHeliumUIScheme ||
+           scheme == url::kFileScheme)) {
         found_scheme = true;
       } else {
         (*text)[semicolon] = ';';
@@ -462,7 +464,8 @@ std::string SegmentURLInternal(std::stri
   // Proceed with about, chrome, and devtools schemes,
   // but not file or nonstandard schemes.
   if ((scheme != url::kAboutScheme) && (scheme != kChromeUIScheme) &&
-      (scheme != kDevToolsScheme) && !url::IsStandard(scheme)) {
+      (scheme != kHeliumUIScheme) && (scheme != kDevToolsScheme) &&
+      !url::IsStandard(scheme)) {
     return scheme;
   }
 
@@ -619,7 +622,9 @@ GURL FixupURLInternal(const std::string&
 
   // For some schemes whose layouts we understand, we rebuild the URL.
   bool chrome_url =
-      (scheme == url::kAboutScheme) || (scheme == kChromeUIScheme);
+      (scheme == url::kAboutScheme)
+      || (scheme == kChromeUIScheme)
+      || (scheme == kHeliumUIScheme);
   bool devtools_url = (scheme == kDevToolsScheme);
   if (chrome_url || devtools_url || url::IsStandard(scheme)) {
     // Replace the about: scheme with the chrome: scheme.
@@ -748,7 +753,11 @@ bool IsEquivalentScheme(const std::strin
                         const std::string& scheme2) {
   return scheme1 == scheme2 ||
          (scheme1 == url::kAboutScheme && scheme2 == kChromeUIScheme) ||
-         (scheme1 == kChromeUIScheme && scheme2 == url::kAboutScheme);
+         (scheme1 == kChromeUIScheme && scheme2 == url::kAboutScheme) ||
+         (scheme1 == kHeliumUIScheme && scheme2 == url::kAboutScheme) ||
+         (scheme1 == url::kAboutScheme && scheme2 == kHeliumUIScheme) ||
+         (scheme1 == kHeliumUIScheme && scheme2 == kChromeUIScheme)   ||
+         (scheme1 == kChromeUIScheme && scheme2 == kHeliumUIScheme);
 }
 
 }  // namespace url_formatter
--- a/chrome/browser/autocomplete/chrome_autocomplete_provider_client.cc
+++ b/chrome/browser/autocomplete/chrome_autocomplete_provider_client.cc
@@ -382,7 +382,7 @@ std::string ChromeAutocompleteProviderCl
 std::string
 ChromeAutocompleteProviderClient::GetEmbedderRepresentationOfAboutScheme()
     const {
-  return content::kChromeUIScheme;
+  return content::kHeliumUIScheme;
 }
 
 std::vector<std::u16string> ChromeAutocompleteProviderClient::GetBuiltinURLs() {
--- a/components/omnibox/browser/builtin_provider.cc
+++ b/components/omnibox/browser/builtin_provider.cc
@@ -73,7 +73,9 @@ void BuiltinProvider::DoBuiltinAutocompl
     // Include some common builtin URLs as the user types the scheme.
     for (const std::u16string& url :
          client_->GetBuiltinsToProvideAsUserTypes()) {
-      AddBuiltinMatch(url, std::u16string(), styles);
+      DCHECK(url.starts_with(u"chrome://"));
+      AddBuiltinMatch(std::u16string(url).replace(0, 6, u"helium"),
+                      std::u16string(), styles);
     }
   } else {
     // Match input about: or |embedderAbout| URL input against builtin URLs.
--- a/chrome/browser/ui/browser_commands.cc
+++ b/chrome/browser/ui/browser_commands.cc
@@ -165,6 +165,7 @@
 #include "components/tabs/public/tab_interface.h"
 #include "components/translate/core/browser/language_state.h"
 #include "components/translate/core/browser/translate_manager.h"
+#include "components/url_formatter/url_formatter.h"
 #include "components/user_education/common/feature_promo/feature_promo_controller.h"
 #include "components/web_modal/web_contents_modal_dialog_manager.h"
 #include "components/webapps/common/web_app_id.h"
@@ -2528,7 +2529,12 @@ bool IsDebuggerAttachedToCurrentTab(Brow
 void CopyURL(BrowserWindowInterface* browser,
              content::WebContents* web_contents) {
   ui::ScopedClipboardWriter scw(ui::ClipboardBuffer::kCopyPaste);
-  scw.WriteText(base::UTF8ToUTF16(web_contents->GetVisibleURL().spec()));
+  const std::u16string formatted_url =
+      url_formatter::FormatUrl(web_contents->GetVisibleURL(),
+                               url_formatter::kFormatReplaceChromeProtocol,
+                               base::UnescapeRule::NORMAL, nullptr,
+                               nullptr, nullptr);
+  scw.WriteText(formatted_url);
 
 #if !BUILDFLAG(IS_ANDROID)
   if (toast_features::IsEnabled(toast_features::kLinkCopiedToast)) {
--- a/chrome/browser/ui/views/tabs/hovercard/hover_card_anchor_target.cc
+++ b/chrome/browser/ui/views/tabs/hovercard/hover_card_anchor_target.cc
@@ -101,7 +101,8 @@ void HoverCardAnchorTarget::SetHoverCard
           url_formatter::kFormatUrlOmitDefaults |
               url_formatter::kFormatUrlOmitHTTPS |
               url_formatter::kFormatUrlOmitTrivialSubdomains |
-              url_formatter::kFormatUrlTrimAfterHost,
+              url_formatter::kFormatUrlTrimAfterHost |
+              url_formatter::kFormatReplaceChromeProtocol,
           base::UnescapeRule::NORMAL, nullptr, nullptr, nullptr);
     }
   }
