--- a/chrome/common/chrome_features.cc
+++ b/chrome/common/chrome_features.cc
@@ -1145,7 +1145,7 @@ BASE_FEATURE(kHttpsFirstBalancedMode, ba
 // Automatically enables HTTPS-First Mode in a balanced configuration when
 // possible.
 BASE_FEATURE(kHttpsFirstBalancedModeAutoEnable,
-             base::FEATURE_DISABLED_BY_DEFAULT);
+             base::FEATURE_ENABLED_BY_DEFAULT);
 
 // Kill switch for crbug.com/1414633.
 BASE_FEATURE(kHttpsFirstModeForAdvancedProtectionUsers,
