--- a/services/network/public/cpp/features.cc
+++ b/services/network/public/cpp/features.cc
@@ -204,7 +204,7 @@ BASE_FEATURE(kIgnoreCorsPreflightPolicy,
 
 // Enables support for the `Variants` response header and reduce
 // accept-language. https://github.com/Tanych/accept-language
-BASE_FEATURE(kReduceAcceptLanguage, base::FEATURE_DISABLED_BY_DEFAULT);
+BASE_FEATURE(kReduceAcceptLanguage, base::FEATURE_ENABLED_BY_DEFAULT);
 
 BASE_FEATURE_PARAM(base::TimeDelta,
                    kReduceAcceptLanguageCacheDuration,
