--- a/chrome/browser/ui/toolbar/pinned_toolbar/pinned_toolbar_actions_model.cc
+++ b/chrome/browser/ui/toolbar/pinned_toolbar/pinned_toolbar_actions_model.cc
@@ -243,11 +243,6 @@ void PinnedToolbarActionsModel::MaybeMig
     UpdatePinnedState(kActionShowChromeLabs, true);
     pref_service_->SetBoolean(prefs::kPinnedChromeLabsMigrationComplete, true);
   }
-  if (features::HasTabSearchToolbarButton() &&
-      !pref_service_->GetBoolean(prefs::kTabSearchMigrationComplete)) {
-    UpdatePinnedState(kActionTabSearch, true);
-    pref_service_->SetBoolean(prefs::kTabSearchMigrationComplete, true);
-  }
   if (!pref_service_->GetBoolean(prefs::kPinnedCastMigrationComplete)) {
     bool previously_pinned =
         pref_service_->GetBoolean(prefs::kShowCastIconInToolbar);
--- a/chrome/browser/ui/toolbar/toolbar_pref_names.cc
+++ b/chrome/browser/ui/toolbar/toolbar_pref_names.cc
@@ -23,14 +23,6 @@ void RegisterProfilePrefs(user_prefs::Pr
     default_pinned_actions.Append(chrome_labs_action.value());
   }
 
-  if (features::HasTabSearchToolbarButton()) {
-    const std::optional<std::string>& tab_search_action =
-        actions::ActionIdMap::ActionIdToString(kActionTabSearch);
-    if (tab_search_action.has_value()) {
-      default_pinned_actions.Append(tab_search_action.value());
-    }
-  }
-
   registry->RegisterListPref(prefs::kPinnedActions,
                              std::move(default_pinned_actions),
                              user_prefs::PrefRegistrySyncable::SYNCABLE_PREF);
--- a/chrome/browser/ui/tab_search_feature.cc
+++ b/chrome/browser/ui/tab_search_feature.cc
@@ -12,6 +12,6 @@ namespace features {
 bool HasTabSearchToolbarButton() {
   // It is important that this value not change at runtime in production. Any
   // future updates to this function must maintain that property.
-  return glic::GlicEnabling::IsEnabledByFlags();
+  return true;
 }
 }  // namespace features
--- a/chrome/browser/ui/views/frame/horizontal_tab_strip_region_view.cc
+++ b/chrome/browser/ui/views/frame/horizontal_tab_strip_region_view.cc
@@ -225,9 +225,7 @@ class TabSearchPositionMetricsLogger {
 HorizontalTabStripRegionView::HorizontalTabStripRegionView(
     BrowserView* browser_view)
     : profile_(browser_view->GetProfile()),
-      render_tab_search_before_tab_strip_(
-          !tabs::GetDefaultTabSearchRightAligned() ||
-          base::FeatureList::IsEnabled(tabs::kHorizontalTabStripComboButton)),
+      render_tab_search_before_tab_strip_(false),
       tab_search_position_metrics_logger_(
           std::make_unique<TabSearchPositionMetricsLogger>(
               browser_view->browser())),
@@ -244,16 +242,6 @@ HorizontalTabStripRegionView::Horizontal
 
   BrowserWindowInterface* const browser = browser_view->browser();
 
-  if (browser &&
-      (browser->GetType() == BrowserWindowInterface::Type::TYPE_NORMAL) &&
-      base::FeatureList::IsEnabled(tabs::kHorizontalTabStripComboButton)) {
-    combo_button_ = AddChildView(std::make_unique<TabStripComboButton>(
-        browser, TabStripComboButton::Context::kHorizontalTabStrip));
-    combo_button_->SetProperty(views::kCrossAxisAlignmentKey,
-                               views::LayoutAlignment::kCenter);
-    combo_button_->MaybeShowIPH();
-  }
-
   if (base::FeatureList::IsEnabled(features::kTabGroupsFocusing)) {
     unfocus_button_ = AddChildView(std::make_unique<TabStripControlButton>(
         browser, views::Button::PressedCallback(), vector_icons::kArrowBackIcon,
@@ -282,26 +270,9 @@ HorizontalTabStripRegionView::Horizontal
 
       tab_strip_action_container->SetProperty(views::kCrossAxisAlignmentKey,
                                               views::LayoutAlignment::kStart);
-    } else if (!base::FeatureList::IsEnabled(
-                   tabs::kHorizontalTabStripComboButton)) {
-      tab_search_button =
-          std::make_unique<TabSearchButton>(browser, Edge::kNone, Edge::kNone);
-      tab_search_button->SetProperty(views::kCrossAxisAlignmentKey,
-                                     views::LayoutAlignment::kCenter);
     }
   }
 
-  if (tab_search_button && render_tab_search_before_tab_strip_) {
-    tab_search_button->SetPaintToLayer();
-    tab_search_button->layer()->SetFillsBoundsOpaquely(false);
-
-    tab_search_button_ = AddChildView(std::move(tab_search_button));
-
-    // Inset between the tabsearch and tabstrip should be reduced to account for
-    // extra spacing.
-    tab_search_button_->SetProperty(views::kViewIgnoredByLayoutKey, true);
-  }
-
   tab_strip_ = AddChildView(CreateTabStrip(this, browser_view));
 
   // Allow the |tab_strip_| to grow into the free space available in
@@ -342,14 +313,6 @@ HorizontalTabStripRegionView::Horizontal
           .WithOrder(3));
 
   SetProperty(views::kElementIdentifierKey, kTabStripRegionElementId);
-
-  if (browser && tab_search_button && !render_tab_search_before_tab_strip_) {
-    tab_search_button_ = AddChildView(std::move(tab_search_button));
-    tab_search_button_->SetProperty(
-        views::kMarginsKey,
-        gfx::Insets::TLBR(0, 0, 0,
-                          GetLayoutConstant(LayoutConstant::kTabStripPadding)));
-  }
   if (tab_strip_action_container) {
     tab_strip_action_container_ =
         AddChildView(std::move(tab_strip_action_container));
