--- a/chrome/browser/prefs/browser_prefs.cc
+++ b/chrome/browser/prefs/browser_prefs.cc
@@ -974,6 +974,12 @@ constexpr char kSafeBrowsingModuleLastCo
 constexpr char kSafeBrowsingModuleOpened[] =
     "safebrowsing.ntp.user_opened_module";
 
+// Deprecated 04/2026, remove on or after 07/2026.
+inline constexpr char kHeliumVerticalCollapsed[] =
+    "helium.browser.vertical_collapsed";
+inline constexpr char kHeliumVerticalUncollapsedWidth[] =
+    "helium.browser.vertical_uncollapsed_width";
+
 // Register local state used only for migration (clearing or moving to a new
 // key).
 void RegisterLocalStatePrefsForMigration(PrefRegistrySimple* registry) {
@@ -1336,6 +1342,10 @@ void RegisterProfilePrefsForMigration(
   registry->RegisterIntegerPref(kSafeBrowsingModuleShownCount, 0);
   registry->RegisterInt64Pref(kSafeBrowsingModuleLastCooldownStartAt, 0);
   registry->RegisterBooleanPref(kSafeBrowsingModuleOpened, false);
+
+  // Deprecated 04/2026.
+  registry->RegisterBooleanPref(kHeliumVerticalCollapsed, false);
+  registry->RegisterIntegerPref(kHeliumVerticalUncollapsedWidth, -1);
 }
 
 }  // namespace
@@ -2620,6 +2630,25 @@ void MigrateObsoleteProfilePrefs(PrefSer
   // Added 03/2026.
   profile_prefs->ClearPref(kNtpPromoPrefLastSnoozed);
 
+  // Added 04/2026.
+  if (!profile_prefs->HasPrefPath(prefs::kVerticalTabsCollapsedState)) {
+    if (const base::Value* user_value =
+            profile_prefs->GetUserPrefValue(kHeliumVerticalCollapsed)) {
+      bool value = user_value->GetBool();
+      profile_prefs->SetBoolean(prefs::kVerticalTabsCollapsedState, value);
+    }
+  }
+  profile_prefs->ClearPref(kHeliumVerticalCollapsed);
+
+  if (!profile_prefs->HasPrefPath(prefs::kVerticalTabsUncollapsedWidth)) {
+    if (const base::Value* user_value =
+            profile_prefs->GetUserPrefValue(kHeliumVerticalUncollapsedWidth)) {
+      int value = user_value->GetInt();
+      profile_prefs->SetInteger(prefs::kVerticalTabsUncollapsedWidth, value);
+    }
+  }
+  profile_prefs->ClearPref(kHeliumVerticalUncollapsedWidth);
+
   // Please don't delete the following line. It is used by PRESUBMIT.py.
   // END_MIGRATE_OBSOLETE_PROFILE_PREFS
 
