--- a/chrome/browser/ui/views/bookmarks/bookmark_button.cc
+++ b/chrome/browser/ui/views/bookmarks/bookmark_button.cc
@@ -63,7 +63,8 @@ BookmarkButtonBase::BookmarkButtonBase(P
   SetImageLabelSpacing(
       GetLayoutConstant(LayoutConstant::kBookmarkBarButtonImageLabelPadding));
 
-  views::InstallPillHighlightPathGenerator(this);
+  views::InstallRoundRectHighlightPathGenerator(this, gfx::Insets(),
+                                                6.0f);
 
   SetFocusBehavior(FocusBehavior::ACCESSIBLE_ONLY);
   views::FocusRing::Get(this)->SetOutsetFocusRingDisabled(true);
--- a/chrome/browser/ui/views/bookmarks/bookmark_menu_button_base.cc
+++ b/chrome/browser/ui/views/bookmarks/bookmark_menu_button_base.cc
@@ -60,7 +60,8 @@ BookmarkMenuButtonBase::BookmarkMenuButt
   ConfigureInkDropForToolbar(this);
   SetImageLabelSpacing(ChromeLayoutProvider::Get()->GetDistanceMetric(
       DISTANCE_RELATED_LABEL_HORIZONTAL_LIST));
-  views::InstallPillHighlightPathGenerator(this);
+  views::InstallRoundRectHighlightPathGenerator(this, gfx::Insets(),
+                                                6.0f);
 
   SetMenuButtonController(std::make_unique<views::MenuButtonController>(
       this, std::move(callback),
