--- a/chrome/browser/ui/window_sizer/window_sizer.cc
+++ b/chrome/browser/ui/window_sizer/window_sizer.cc
@@ -355,9 +355,10 @@ gfx::Rect WindowSizer::GetDefaultWindowB
     default_height = (default_width / 4) * 3;
   }
 
-  return gfx::Rect(kWindowTilePixels + work_area.x(),
-                   kWindowTilePixels + work_area.y(), default_width,
-                   default_height);
+  int center_x = (work_area.width() - default_width) / 2;
+  int center_y = (work_area.height() - default_height) / 2;
+
+  return gfx::Rect(center_x, center_y, default_width, default_height);
 }
 
 void WindowSizer::AdjustBoundsToBeVisibleOnDisplay(
