--- a/chrome/browser/ui/views/tabs/groups/tab_group_editor_bubble_view.cc
+++ b/chrome/browser/ui/views/tabs/groups/tab_group_editor_bubble_view.cc
@@ -138,9 +138,6 @@ constexpr int kSeparatorPadding = 8;
 constexpr int kDialogWidth = 240;
 // The default size of icons that are displayed in the tab group editor bubble.
 static constexpr int kDefaultIconSize = 20;
-// The maximum number of times we will show the footer section with the learn
-// more link.
-constexpr int kFooterDisplayLimit = 5;
 
 int GetHorizontalSpacing() {
   return ChromeLayoutProvider::Get()->GetDistanceMetric(
@@ -505,10 +502,7 @@ bool TabGroupEditorBubbleView::IsGroupSh
 }
 
 bool TabGroupEditorBubbleView::ShouldShowSavedFooter() const {
-  PrefService* pref_service = browser_->profile()->GetPrefs();
-  return (CanSaveGroups() && pref_service &&
-          saved_tab_group_prefs::GetLearnMoreFooterShownCount(pref_service) <
-              kFooterDisplayLimit);
+  return false;
 }
 
 bool TabGroupEditorBubbleView::OwnsGroup() const {
