--- a/chrome/browser/resources/side_panel/customize_chrome/app.ts
+++ b/chrome/browser/resources/side_panel/customize_chrome/app.ts
@@ -130,8 +130,6 @@ export class AppElement extends AppEleme
     this.listenerIds_ = [
       this.apiProxy_.callbackRouter.scrollToSection.addListener(
           this.onScrollToSection_.bind(this)),
-      this.apiProxy_.callbackRouter.attachedTabStateUpdated.addListener(
-          this.onAttachedTabStateUpdated_.bind(this)),
       this.apiProxy_.callbackRouter.setThemeEditable.addListener(
           this.onSetThemeEditable_.bind(this)),
       this.apiProxy_.callbackRouter.setFooterSettings.addListener(
@@ -218,22 +216,6 @@ export class AppElement extends AppEleme
     element.scrollIntoView({behavior: 'auto'});
   }
 
-
-  private onAttachedTabStateUpdated_(newTabPageType: NewTabPageType) {
-    if (this.newTabPageType_ === newTabPageType) {
-      return;
-    }
-
-    this.newTabPageType_ = newTabPageType;
-
-    // Since some pages aren't supported in non first party mode,
-    // change the section back to the overview.
-    if (!this.isSourceTabFirstPartyNtp_() &&
-        !this.pageSupportedOnNonFirstPartyNtps()) {
-      this.page_ = CustomizeChromePage.OVERVIEW;
-    }
-  }
-
   private onSetThemeEditable_(isThemeEditable: boolean) {
     this.showEditTheme_ = isThemeEditable;
   }
@@ -352,10 +334,6 @@ export class AppElement extends AppEleme
     await this.updateComplete;
     page.focusOnBackButton();
   }
-
-  private pageSupportedOnNonFirstPartyNtps() {
-    return this.page_ === CustomizeChromePage.TOOLBAR;
-  }
 }
 
 declare global {
--- a/chrome/browser/resources/side_panel/customize_chrome/appearance.ts
+++ b/chrome/browser/resources/side_panel/customize_chrome/appearance.ts
@@ -129,10 +129,6 @@ export class AppearanceElement extends A
       this.apiProxy_.callbackRouter.setTheme.addListener((theme: Theme) => {
         this.theme_ = theme;
       }),
-      this.apiProxy_.callbackRouter.attachedTabStateUpdated.addListener(
-          (newTabPageType: NewTabPageType) => {
-            this.newTabPageType_ = newTabPageType;
-          }),
       this.apiProxy_.callbackRouter.setThemeEditable.addListener(
           (isThemeEditable: boolean) => {
             this.showEditTheme_ = isThemeEditable;
