--- a/components/qr_code_generator/bitmap_generator.cc
+++ b/components/qr_code_generator/bitmap_generator.cc
@@ -334,7 +334,9 @@ base::expected<SkBitmap, Error> Generate
     // The QR version (i.e. size) must be >= 5 because otherwise the dino
     // painted over the middle covers too much of the code to be decodable.
     constexpr int kMinimumQRVersion = 5;
-    auto qr_result = GenerateCode(data, kMinimumQRVersion);
+    auto qr_result = center_image == CenterImage::kNoCenterImage
+                         ? GenerateCode(data)
+                         : GenerateCode(data, kMinimumQRVersion);
     if (!qr_result.has_value()) {
       return base::unexpected(qr_result.error());
     }
--- a/chrome/browser/ui/views/qrcode_generator/qrcode_generator_bubble.cc
+++ b/chrome/browser/ui/views/qrcode_generator/qrcode_generator_bubble.cc
@@ -159,9 +159,9 @@ void QRCodeGeneratorBubble::UpdateQRCont
     qr_code = base::unexpected(qrcode_error_override_.value());
   } else {
     qr_code = qr_code_generator::GenerateImage(
-        base::as_byte_span(input), qr_code_generator::ModuleStyle::kCircles,
-        qr_code_generator::LocatorStyle::kRounded,
-        qr_code_generator::CenterImage::kDino,
+        base::as_byte_span(input), qr_code_generator::ModuleStyle::kSquares,
+        qr_code_generator::LocatorStyle::kSquare,
+        qr_code_generator::CenterImage::kNoCenterImage,
         qr_code_generator::QuietZone::kIncluded);
   }
 
