--- a/chrome/browser/ui/color/material_chrome_color_mixer.cc
+++ b/chrome/browser/ui/color/material_chrome_color_mixer.cc
@@ -50,6 +50,9 @@ void ApplyDefaultChromeRefreshToolbarCol
 
 void AddMaterialChromeColorMixer(ui::ColorProvider* provider,
                                  const ui::ColorProviderKey& key) {
+  const bool dark_mode =
+      key.color_mode == ui::ColorProviderKey::ColorMode::kDark;
+
   // Adds the color recipes for browser UI colors (toolbar, bookmarks bar,
   // downloads bar etc). While both design systems continue to exist, the
   // material recipes are intended to leverage the existing chrome color mixers,
@@ -400,7 +403,8 @@ void AddMaterialChromeColorMixer(ui::Col
       ui::kColorSysOnSurfaceSubtle};
 
   // Toolbar colors.
-  mixer[kColorToolbar] = {ui::kColorSysBase};
+  mixer[kColorToolbar] = {dark_mode ? ui::kColorFrameActive
+                                    : ui::kColorSysBase};
   mixer[kColorToolbarButtonBackgroundHighlightedDefault] = {
       ui::kColorSysStateHoverOnSubtle};
   mixer[kColorToolbarButtonBorder] = {ui::kColorSysOutline};
@@ -412,7 +416,10 @@ void AddMaterialChromeColorMixer(ui::Col
       {ui::kColorSysStateDisabled}, {kColorToolbar})};
   mixer[kColorToolbarButtonIconPressed] = {kColorToolbarButtonIconHovered};
   mixer[kColorToolbarButtonText] = {ui::kColorSysOnSurfaceSecondary};
-  mixer[kColorToolbarContentAreaSeparator] = {ui::kColorSysSurfaceVariant};
+  mixer[kColorToolbarContentAreaSeparator] = {
+      dark_mode ? ui::GetResultingPaintColor({ui::kColorSysStateHoverOnSubtle},
+                                             {kColorToolbar})
+                : ui::kColorSysSurfaceVariant};
   mixer[kColorToolbarFeaturePromoHighlight] = {ui::kColorSysPrimary};
   mixer[kColorToolbarIconContainerBorder] = {ui::kColorSysNeutralOutline};
   mixer[kColorToolbarInkDropHover] = {ui::kColorSysStateHoverOnSubtle};
@@ -424,8 +431,11 @@ void AddMaterialChromeColorMixer(ui::Col
   mixer[kColorToolbarExtensionSeparatorDisabled] = {
       kColorToolbarButtonIconInactive};
   mixer[kColorToolbarSeparator] = {kColorToolbarSeparatorDefault};
+  const auto toolbar_separator_bg = dark_mode
+                                        ? ui::ColorTransform(ui::kColorSysBase)
+                                        : ui::ColorTransform(kColorToolbar);
   mixer[kColorToolbarSeparatorDefault] =
-      ui::AlphaBlend(kColorToolbarButtonIcon, kColorToolbar, 0x3A);
+      ui::AlphaBlend(kColorToolbarButtonIcon, toolbar_separator_bg, 0x3A);
   mixer[kColorToolbarText] = {kColorToolbarTextDefault};
   mixer[kColorToolbarTextDefault] = {ui::kColorSysOnSurfaceSecondary};
   mixer[kColorToolbarTextDisabled] = {kColorToolbarTextDisabledDefault};
--- a/chrome/browser/ui/color/chrome_color_mixer.cc
+++ b/chrome/browser/ui/color/chrome_color_mixer.cc
@@ -117,9 +117,13 @@ ui::ColorTransform GetToolbarTopSeparato
 // Apply updates to the Omnibox background color tokens per GM3 spec.
 void ApplyGM3OmniboxBackgroundColor(ui::ColorMixer& mixer,
                                     const ui::ColorProviderKey& key) {
+  const bool dark_mode =
+      key.color_mode == ui::ColorProviderKey::ColorMode::kDark;
   // Apply omnibox background color updates only to non-themed clients.
   if (!key.custom_theme) {
-    mixer[kColorLocationBarBackground] = {ui::kColorSysOmniboxContainer};
+    mixer[kColorLocationBarBackground] = {
+        dark_mode ? ui::ColorTransform(kColorOmniboxResultsBackground)
+                  : ui::ColorTransform(ui::kColorSysOmniboxContainer)};
     mixer[kColorLocationBarBackgroundHovered] =
         ui::GetResultingPaintColor(ui::kColorSysStateHoverBrightBlendProtection,
                                    kColorLocationBarBackground);
@@ -810,8 +814,10 @@ void AddChromeColorMixer(ui::ColorProvid
       ui::SetAlpha(kColorToolbarButtonIcon, gfx::kGoogleGreyAlpha500)};
   mixer[kColorToolbarButtonIconPressed] = {kColorToolbarButtonIconHovered};
   mixer[kColorToolbarButtonText] = ui::GetColorWithMaxContrast(kColorToolbar);
-  mixer[kColorToolbarContentAreaSeparator] =
-      ui::AlphaBlend(kColorToolbarButtonIcon, kColorToolbar, 0x3A);
+  mixer[kColorToolbarContentAreaSeparator] = {
+      dark_mode ? ui::GetResultingPaintColor({ui::kColorSysStateHoverOnSubtle},
+                                             {kColorToolbar})
+                : ui::kColorSysSurfaceVariant};
   mixer[kColorToolbarFeaturePromoHighlight] =
       ui::PickGoogleColor(ui::kColorAccent, kColorToolbar,
                           color_utils::kMinimumVisibleContrastRatio);
--- a/ui/color/ui_color_mixer.cc
+++ b/ui/color/ui_color_mixer.cc
@@ -31,7 +31,8 @@ void AddUiColorMixer(ColorProvider* prov
   mixer[kColorBadgeInCocoaMenuBackground] = {kColorBadgeBackground};
   mixer[kColorBadgeInCocoaMenuForeground] = {kColorBadgeForeground};
   mixer[kColorBubbleBackground] = {kColorPrimaryBackground};
-  mixer[kColorBubbleBorder] = {kColorSysSurfaceVariant};
+  mixer[kColorBubbleBorder] = {dark_mode ? ui::kColorSysStateHoverOnSubtle
+                                         : ui::kColorSysSurfaceVariant};
   mixer[kColorBubbleBorderShadowLarge] = {SetAlpha(kColorShadowBase, 0x1A)};
   mixer[kColorBubbleBorderShadowSmall] = {SetAlpha(kColorShadowBase, 0x33)};
   mixer[kColorBubbleFooterBackground] = {kColorSubtleEmphasisBackground};
--- a/chrome/browser/ui/color/material_tab_strip_color_mixer.cc
+++ b/chrome/browser/ui/color/material_tab_strip_color_mixer.cc
@@ -17,6 +17,10 @@ constexpr SkAlpha kWebUiTabStripScrollba
 
 /* 16% opacity */
 constexpr SkAlpha kWebUiTabStripTabSeparatorAlpha = 0.16 * 255;
+
+/* 50% opacity */
+constexpr SkAlpha kTabInactiveHoverAlpha = 0.5 * 255;
+
 }  // namespace
 
 void AddMaterialTabStripColorMixer(ui::ColorProvider* provider,
@@ -33,15 +37,17 @@ void AddMaterialTabStripColorMixer(ui::C
   ui::ColorMixer& mixer = provider->AddMixer();
   mixer[kColorDetachedTabBackgroundActiveFrameActive] = {
       dark_mode ? ui::kColorSysSurfaceVariant : ui::kColorSysBase};
-  mixer[kColorTabBackgroundActiveFrameActive] = {ui::kColorSysBase};
+  mixer[kColorTabBackgroundActiveFrameActive] = {kColorLocationBarBackground};
   mixer[kColorTabBackgroundActiveFrameInactive] = {
-      kColorTabBackgroundActiveFrameActive};
+      kColorLocationBarBackground};
 
-  mixer[kColorTabBackgroundInactiveFrameActive] = {ui::kColorSysHeader};
-  mixer[kColorTabBackgroundInactiveFrameInactive] = {
-      ui::kColorSysHeaderInactive};
+  mixer[kColorTabBackgroundInactiveFrameActive] = ui::SetAlpha(
+      kColorTabBackgroundActiveFrameActive, SK_AlphaTRANSPARENT);
+  mixer[kColorTabBackgroundInactiveFrameInactive] = ui::SetAlpha(
+      kColorTabBackgroundActiveFrameInactive, SK_AlphaTRANSPARENT);
   mixer[kColorTabBackgroundInactiveHoverFrameActive] = {
-      ui::kColorSysStateHeaderHover};
+      ui::SetAlpha(kColorTabBackgroundActiveFrameActive,
+                   kTabInactiveHoverAlpha)};
   mixer[kColorTabStripComboButtonSeparator] = {ui::kColorSysDivider};
   mixer[kColorTabStripControlButtonInkDrop] = {ui::kColorSysStateHeaderHover};
   mixer[kColorTabStripControlButtonInkDropRipple] = {
@@ -49,19 +55,16 @@ void AddMaterialTabStripColorMixer(ui::C
 
   // TODO(tbergquist): Use kColorSysStateHeaderHoverInactive, once it exists.
   mixer[kColorTabBackgroundInactiveHoverFrameInactive] = {
-      ui::kColorSysStateHeaderHoverInactive};
+      ui::SetAlpha(kColorTabBackgroundActiveFrameInactive,
+                   kTabInactiveHoverAlpha)};
 
-  mixer[kColorTabBackgroundSelectedFrameActive] = {ui::GetResultingPaintColor(
-      ui::kColorSysStateHeaderSelect, kColorTabBackgroundInactiveFrameActive)};
+  mixer[kColorTabBackgroundSelectedFrameActive] = {kColorLocationBarBackground};
   mixer[kColorTabBackgroundSelectedFrameInactive] = {
-      ui::GetResultingPaintColor(ui::kColorSysStateHeaderSelect,
-                                 kColorTabBackgroundInactiveFrameInactive)};
+      kColorLocationBarBackground};
   mixer[kColorTabBackgroundSelectedHoverFrameActive] = {
-      ui::GetResultingPaintColor(ui::kColorSysStateHoverDimBlendProtection,
-                                 kColorTabBackgroundSelectedFrameActive)};
+      kColorLocationBarBackground};
   mixer[kColorTabBackgroundSelectedHoverFrameInactive] = {
-      ui::GetResultingPaintColor(ui::kColorSysStateHoverDimBlendProtection,
-                                 kColorTabBackgroundSelectedFrameInactive)};
+      kColorLocationBarBackground};
 #if !BUILDFLAG(IS_ANDROID)
   mixer[kColorTabDiscardRingFrameActive] = {ui::kColorSysStateInactiveRing};
   mixer[kColorTabDiscardRingFrameInactive] = {kColorTabDiscardRingFrameActive};
@@ -69,12 +72,13 @@ void AddMaterialTabStripColorMixer(ui::C
   mixer[kColorTabForegroundActiveFrameActive] = {ui::kColorSysOnSurface};
   mixer[kColorTabForegroundActiveFrameInactive] = {
       kColorTabForegroundActiveFrameActive};
-  mixer[kColorTabForegroundInactiveFrameActive] =
-      ui::BlendForMinContrast({ui::kColorSysOnSurfaceSecondary},
-                              {kColorTabBackgroundInactiveFrameActive});
-  mixer[kColorTabForegroundInactiveFrameInactive] =
-      ui::BlendForMinContrast({kColorTabForegroundInactiveFrameActive},
-                              {kColorTabBackgroundInactiveFrameInactive});
+  mixer[kColorTabForegroundInactiveFrameActive] = {
+      ui::kColorSysOnSurfaceSecondary};
+  mixer[kColorTabForegroundInactiveFrameInactive] = {
+      kColorTabForegroundInactiveFrameActive};
+  mixer[kColorTabCloseButtonFocusRingInactive] = {
+      ui::kColorSysStateFocusRing};
+  mixer[kColorTabFocusRingInactive] = {ui::kColorSysStateFocusRing};
 
   /* WebUI Tab Strip colors. */
   mixer[kColorWebUiTabStripBackground] = {ui::kColorSysHeader};
@@ -98,10 +102,8 @@ void AddMaterialTabStripColorMixer(ui::C
       ui::kColorSysOnSurfaceSubtle};
   mixer[kColorNewTabButtonCRForegroundFrameInactive] = {
       ui::kColorSysOnSurfaceSubtle};
-  mixer[kColorNewTabButtonCRBackgroundFrameActive] = {
-      ui::kColorSysHeaderContainer};
-  mixer[kColorNewTabButtonCRBackgroundFrameInactive] = {
-      ui::kColorSysHeaderContainerInactive};
+  mixer[kColorNewTabButtonCRBackgroundFrameActive] = {kColorToolbar};
+  mixer[kColorNewTabButtonCRBackgroundFrameInactive] = {kColorToolbar};
 
   mixer[kColorTabSearchButtonCRForegroundFrameActive] = {
       ui::kColorSysOnSurfacePrimary};
