--- a/chrome/browser/ui/views/overlay/video_overlay_window_views.cc
+++ b/chrome/browser/ui/views/overlay/video_overlay_window_views.cc
@@ -1252,6 +1252,7 @@ void VideoOverlayWindowViews::SetUpViews
                           base::Unretained(this)));
   live_caption_button->SetSize(kActionButtonSize);
   live_caption_button->SetIsLiveCaptionDialogOpen(false);
+  live_caption_button->SetVisible(false);
 
   auto live_caption_dialog = std::make_unique<OverlayWindowLiveCaptionDialog>(
       Profile::FromBrowserContext(
@@ -2426,8 +2427,8 @@ void VideoOverlayWindowViews::OnLiveCapt
   SetLiveCaptionDialogVisibility(!live_caption_dialog_->GetVisible());
 }
 
-void VideoOverlayWindowViews::SetLiveCaptionDialogVisibility(
-    bool wanted_visibility) {
+void VideoOverlayWindowViews::SetLiveCaptionDialogVisibility(bool) {
+  const bool wanted_visibility = false;
   if (wanted_visibility == live_caption_dialog_->GetVisible()) {
     return;
   }
