--- a/chrome/browser/ui/views/intent_picker/intent_picker_view_page_action_controller.cc
+++ b/chrome/browser/ui/views/intent_picker/intent_picker_view_page_action_controller.cc
@@ -37,30 +37,7 @@ void IntentPickerViewPageActionControlle
       tab_interface_->GetTabFeatures()->page_action_controller();
   CHECK(page_action_controller);
   if (should_show_icon) {
-    if (apps::features::ShouldShowLinkCapturingUX()) {
-      // If link capturing is enabled, override the icon, text and tooltip
-      // based upon the navigated website.
-      if (app_icon.IsEmpty()) {
-        page_action_controller->ClearOverrideImage(kActionShowIntentPicker);
-      } else {
-        page_action_controller->OverrideImage(kActionShowIntentPicker,
-                                              app_icon);
-      }
-      page_action_controller->OverrideText(
-          kActionShowIntentPicker,
-          l10n_util::GetStringUTF16(IDS_INTENT_CHIP_OPEN_IN_APP));
-      page_action_controller->OverrideTooltip(
-          kActionShowIntentPicker,
-          l10n_util::GetStringUTF16(IDS_INTENT_CHIP_OPEN_IN_APP));
-      page_action_controller->Show(kActionShowIntentPicker);
-
-      if (should_show_expanded_chip) {
-        page_action_controller->ShowSuggestionChip(kActionShowIntentPicker,
-                                                   {.should_animate = false});
-      }
-    } else {
-      page_action_controller->Show(kActionShowIntentPicker);
-    }
+    page_action_controller->Show(kActionShowIntentPicker);
   } else {
     HideIcon();
   }
--- a/chrome/browser/ui/views/page_action/page_action_view.cc
+++ b/chrome/browser/ui/views/page_action/page_action_view.cc
@@ -55,7 +55,7 @@ PageActionView::PageActionView(actions::
 
   SetVisible(false);
   label()->SetVisible(false);
-  SetUseTonalColorsWhenExpanded(true);
+  SetUseTonalColorsWhenExpanded(false);
   SetBackgroundVisibility(BackgroundVisibility::kWithLabel);
   UpdateBorder();
   SetExpandedLabelAdditionalInsets(views::Inset1D(4, 8));
@@ -203,6 +203,10 @@ void PageActionView::UpdateBorder() {
   SetBorder(views::CreateEmptyBorder(border_insets));
 }
 
+bool PageActionView::ShouldShowLabel() const {
+  return false;
+}
+
 bool PageActionView::ShouldShowSeparator() const {
   return false;
 }
--- a/chrome/browser/ui/views/page_action/page_action_view.h
+++ b/chrome/browser/ui/views/page_action/page_action_view.h
@@ -82,6 +82,7 @@ class PageActionView : public IconLabelB
   void OnThemeChanged() override;
   void OnTouchUiChanged() override;
   void UpdateBorder() override;
+  bool ShouldShowLabel() const override;
   bool ShouldShowLabelAfterAnimation() const override;
   bool ShouldShowSeparator() const override;
   bool ShouldUpdateInkDropOnClickCanceled() const override;
