--- a/chrome/browser/ui/views/location_bar/icon_label_bubble_view.cc
+++ b/chrome/browser/ui/views/location_bar/icon_label_bubble_view.cc
@@ -51,7 +51,7 @@ constexpr int kIconLabelBubbleSeparatorW
 
 // Amount of space on either side of the separator that appears after the icon
 // or label.
-constexpr int kIconLabelBubbleSpaceBesideSeparator = 8;
+constexpr int kIconLabelBubbleSpaceBesideSeparator = 6;
 
 // The length of the separator's fade animation. These values are empirical.
 constexpr int kIconLabelBubbleFadeInDurationMs = 250;
@@ -87,9 +87,7 @@ void IconLabelBubbleView::SeparatorView:
   const SkColor separator_color = SkColorSetA(
       owner_->delegate_->GetIconLabelBubbleSurroundingForegroundColor(),
       owner_->delegate_->GetIconLabelBubbleSeparatorAlpha());
-  const float x = GetLocalBounds().right() -
-                  owner_->GetEndPaddingWithSeparator() -
-                  1.0f / canvas->image_scale();
+  const float x = 0.5f;
   canvas->DrawLine(gfx::PointF(x, GetLocalBounds().y()),
                    gfx::PointF(x, GetLocalBounds().bottom()), separator_color);
 }
@@ -288,8 +286,7 @@ views::ProposedLayout IconLabelBubbleVie
   // value to be, since it won't be visible.
   const int label_x = image_bounds_with_label.right() + GetInternalSpacing();
   const int available_label_width =
-      std::max(0, layout.host_size.width() - label_x - bubble_trailing_padding -
-                      GetWidthBetweenIconAndSeparator());
+      std::max(0, layout.host_size.width() - label_x - bubble_trailing_padding);
   gfx::Rect label_bounds(label_x, 0, available_label_width, height);
   layout.child_layouts.emplace_back(
       label(),
@@ -312,10 +309,9 @@ views::ProposedLayout IconLabelBubbleVie
   gfx::Rect separator_bounds(label_bounds);
   separator_bounds.Inset(
       gfx::Insets::VH((separator_bounds.height() - separator_height) / 2, 0));
-  float separator_width =
-      GetWidthBetweenIconAndSeparator() + GetEndPaddingWithSeparator();
+  float separator_width = 2;
   int separator_x =
-      label()->GetText().empty() ? image_bounds.right() : label_bounds.right();
+      (label()->GetText().empty() ? image_bounds.right() : label_bounds.right()) + 2;
 
   layout.child_layouts.emplace_back(
       const_cast<views::View*>(this->separator_view()), ShouldShowSeparator(),
@@ -627,7 +623,9 @@ gfx::RoundedCornersF IconLabelBubbleView
   if (radii_.has_value()) {
     return radii_.value();
   }
-  return gfx::RoundedCornersF(GetPreferredSize().height() / 2);
+
+  // location bar rounding is 8px and padding is 2px, so inner rounding is 6px
+  return gfx::RoundedCornersF(6.0f);
 }
 
 void IconLabelBubbleView::SetCornerRadii(const gfx::RoundedCornersF& radii) {
@@ -675,7 +673,7 @@ int IconLabelBubbleView::GetInternalSpac
   }
 
   constexpr int kDefaultInternalSpacingTouchUI = 10;
-  constexpr int kDefaultInternalSpacingChromeRefresh = 4;
+  constexpr int kDefaultInternalSpacingChromeRefresh = 7;
 
   return (ui::TouchUiController::Get()->touch_ui()
               ? kDefaultInternalSpacingTouchUI
@@ -820,12 +818,12 @@ SkPath IconLabelBubbleView::GetHighlight
   highlight_bounds = GetMirroredRect(highlight_bounds);
 
   const SkRect rect = RectToSkRect(highlight_bounds);
-  gfx::RoundedCornersF radii = GetCornerRadii();
-  const SkVector sk_radii[4] = {{radii.upper_left(), radii.upper_left()},
-                                {radii.upper_right(), radii.upper_right()},
-                                {radii.lower_right(), radii.lower_right()},
-                                {radii.lower_left(), radii.lower_left()}};
-
+  const float layout_radius = static_cast<float>(
+      GetLayoutConstant(LayoutConstant::kLocationBarChildCornerRadius));
+  const SkVector sk_radii[4] = {{layout_radius, layout_radius},
+                                  {layout_radius, layout_radius},
+                                  {layout_radius, layout_radius},
+                                  {layout_radius, layout_radius}};
   return SkPath::RRect(SkRRect::MakeRectRadii(rect, sk_radii));
 }
 
--- a/chrome/browser/ui/views/location_bar/location_bar_view.cc
+++ b/chrome/browser/ui/views/location_bar/location_bar_view.cc
@@ -215,10 +215,10 @@ int IncrementalMinimumWidth(const views:
 }
 
 // The padding between the intent chip and the other trailing decorations.
-constexpr int kIntentChipIntraItemPadding = 12;
+constexpr int kIntentChipIntraItemPadding = 4;
 
 // The padding between the content setting icons and other trailing decorations.
-constexpr int kContentSettingIntraItemPadding = 8;
+constexpr int kContentSettingIntraItemPadding = kIntentChipIntraItemPadding;
 
 }  // namespace
 
@@ -264,7 +264,10 @@ LocationBarView::LocationBarView(Browser
                  !v->GetOmniboxController()->IsPopupOpen();
         }));
     views::FocusRing::Get(this)->SetOutsetFocusRingDisabled(true);
-    views::InstallPillHighlightPathGenerator(this);
+
+    // Same radius as the location bar.
+    views::InstallRoundRectHighlightPathGenerator(
+      this, gfx::Insets(), GetBorderRadius());
 
 #if BUILDFLAG(OS_LEVEL_GEOLOCATION_PERMISSION_SUPPORTED)
     if (features::IsOsLevelGeolocationPermissionSupportEnabled()) {
@@ -324,9 +327,9 @@ void LocationBarView::Init() {
       CONTEXT_OMNIBOX_PRIMARY, views::style::STYLE_PRIMARY);
 
   const gfx::FontList& omnibox_chip_font_list = typography_provider.GetFont(
-      CONTEXT_OMNIBOX_PRIMARY, views::style::STYLE_BODY_4_EMPHASIS);
+      CONTEXT_OMNIBOX_PRIMARY, views::style::STYLE_PRIMARY);
   const gfx::FontList& page_action_font_list = typography_provider.GetFont(
-      CONTEXT_OMNIBOX_PRIMARY, views::style::STYLE_BODY_3_EMPHASIS);
+      CONTEXT_OMNIBOX_PRIMARY, views::style::STYLE_PRIMARY);
 
   auto location_icon_view =
       std::make_unique<LocationIconView>(omnibox_chip_font_list, this, this);
@@ -482,7 +485,7 @@ void LocationBarView::Init() {
   // if all page actions (i.e. up to bookmark star) are migrated.
   const bool should_bridge_containers =
       !IsPageActionMigrated(PageActionIconType::kBookmarkStar);
-  static constexpr int kBetweenIconSpacing = 8;
+  static constexpr int kBetweenIconSpacing = kIntentChipIntraItemPadding;
   const page_actions::PageActionViewParams page_action_params{
       .icon_size =
           GetLayoutConstant(LayoutConstant::kLocationBarTrailingIconSize),
@@ -509,7 +512,6 @@ void LocationBarView::Init() {
     if (optimization_guide::features::ShouldEnableOptimizationGuideIconView()) {
       params.types_enabled.push_back(PageActionIconType::kOptimizationGuide);
     }
-    params.types_enabled.push_back(PageActionIconType::kManagePasswords);
     if (!apps::features::ShouldShowLinkCapturingUX()) {
       params.types_enabled.push_back(PageActionIconType::kIntentPicker);
     }
@@ -518,53 +520,9 @@ void LocationBarView::Init() {
     params.types_enabled.push_back(PageActionIconType::kZoom);
     params.types_enabled.push_back(PageActionIconType::kFileSystemAccess);
 
-    params.types_enabled.push_back(PageActionIconType::kCookieControls);
-    params.types_enabled.push_back(
-        PageActionIconType::kPaymentsOfferNotification);
     params.types_enabled.push_back(PageActionIconType::kMemorySaver);
     params.types_enabled.push_back(PageActionIconType::kFederation);
   }
-  params.types_enabled.push_back(PageActionIconType::kSaveCard);
-  params.types_enabled.push_back(PageActionIconType::kSaveIban);
-  params.types_enabled.push_back(PageActionIconType::kFilledCardInformation);
-  params.types_enabled.push_back(PageActionIconType::kVirtualCardEnroll);
-  params.types_enabled.push_back(PageActionIconType::kMandatoryReauth);
-
-  if (browser_ && lens::features::IsOmniboxEntryPointEnabled()) {
-    // The persistent compact entrypoint should be positioned directly before
-    // the star icon and the prominent expanding entrypoint should be
-    // positioned in the leading position. This entrypoint will be suppressed
-    // if the AIM page action is enabled and the user meets AIM eligibility
-    // criteria, since we want to avoid both showing up when the user focuses
-    // the Omnibox.
-    if (lens::features::IsOmniboxEntrypointAlwaysVisible()) {
-      params.types_enabled.push_back(PageActionIconType::kLensOverlay);
-    } else {
-      params.types_enabled.insert(params.types_enabled.begin(),
-                                  PageActionIconType::kLensOverlay);
-    }
-  }
-
-  if (browser_ && lens::features::IsLensOverlayEduActionChipEnabled()) {
-    // Position in the leading position, like the expanding entrypoint for
-    // kLensOverlay above. While both chips may be enabled, they will not appear
-    // at the same time due to different focus behavior.
-    params.types_enabled.insert(params.types_enabled.begin(),
-                                PageActionIconType::kLensOverlayHomework);
-  }
-
-  // Because AIM eligibility can change during the lifecycle of the
-  // `LocationBarView`, the AI Mode page action is added regardless of
-  // eligibility, but its visibility is toggled to match eligibility.
-  if (base::FeatureList::IsEnabled(omnibox::kAiModeOmniboxEntryPoint)) {
-    // Position in the leading position, like the entrypoint for
-    // kLensOverlayHomework above. While both chips may be enabled, they will
-    // not appear at the same time due to different focus behavior. The
-    // visibility of this entrypoint is dependent on whether or not the user
-    // meets AIM eligibility criteria.
-    params.types_enabled.insert(params.types_enabled.begin(),
-                                PageActionIconType::kAiMode);
-  }
 
   if (browser_ && !is_popup_mode_) {
     params.types_enabled.push_back(PageActionIconType::kBookmarkStar);
@@ -581,17 +539,6 @@ void LocationBarView::Init() {
       AddChildView(std::make_unique<PageActionIconContainerView>(params));
   page_action_icon_controller_ = page_action_icon_container_->controller();
 
-  if (!page_action_icon_container_->children().empty() &&
-      base::FeatureList::IsEnabled(omnibox::kAiModeOmniboxEntryPoint) &&
-      !IsPageActionMigrated(PageActionIconType::kAiMode)) {
-    auto* first_page_action_icon_view = static_cast<PageActionIconView*>(
-        page_action_icon_container_->children().front());
-    DCHECK(first_page_action_icon_view->action_id() == kActionAiMode)
-        << "kActionAiMode must be the first child in "
-           "PageActionIconContainerView to ensure it's the left-most page "
-           "action.";
-  }
-
   auto clear_all_button = views::CreateVectorImageButton(base::BindRepeating(
       static_cast<void (OmniboxView::*)(const std::u16string&)>(
           &OmniboxView::SetUserText),
@@ -621,7 +568,7 @@ int LocationBarView::GetBorderRadius() c
 // static
 int LocationBarView::ComputeBorderRadius(gfx::Size size) {
   return ChromeLayoutProvider::Get()->GetCornerRadiusMetric(
-      views::Emphasis::kMaximum, size);
+      views::Emphasis::kHigh, size);
 }
 
 // static
@@ -840,9 +787,9 @@ void LocationBarView::Layout(PassKey) {
   // TODO(manukh): Move constants to layout_constants.cc.
   // The padding between the left edges of the location bar and the LHS icon,
   // e.g. the page info icon, the google G icon, the selected suggestion icon.
-  int icon_left = 5;
+  int icon_left = 2;
   // The padding between the LHS icon and the text.
-  int text_left = 8;
+  int text_left = 5;
 
   // Apply indentation to align the omnibox input icon and the text with those
   // of the suggestions in the popup. However, there are two exceptions where
@@ -861,22 +808,22 @@ void LocationBarView::Layout(PassKey) {
                              !location_icon_view_->ShouldShowLabel() &&
                              !ShouldShowKeywordBubble();
   if (should_indent) {
-    icon_left += 7 /*icon_indent*/;
+    icon_left += 0 /*icon_indent*/;
     icon_left += omnibox_feature_configs::AdjustOmniboxIndent()
                      .Get()
                      .input_icon_indent_offset;
-    text_left += 6 /*text_indent*/;
+    text_left += 0 /*text_indent*/;
     text_left += omnibox_feature_configs::AdjustOmniboxIndent()
                      .Get()
                      .input_text_indent_offset;
   } else if (ShouldShowKeywordBubble()) {
     // Otherwise, if in keyword mode, adjust indentation to align the icon and
     // the text with the suggestion icons & texts.
-    icon_left += 9; /*icon_indent_keyword_mode*/
+    icon_left += 0; /*icon_indent_keyword_mode*/
     icon_left += omnibox_feature_configs::AdjustOmniboxIndent()
                      .Get()
                      .input_icon_indent_offset;
-    text_left += -9; /*text_indent_keyword_mode*/
+    text_left += 0; /*text_indent_keyword_mode*/
   }
 
   LocationBarLayout leading_decorations(LocationBarLayout::Position::kLeftEdge,
@@ -1547,7 +1494,7 @@ int LocationBarView::GetMinimumTrailingW
 }
 
 gfx::Rect LocationBarView::GetLocalBoundsWithoutEndcaps() const {
-  const int border_radius = height() / 2;
+  const int border_radius = GetBorderRadius();
   gfx::Rect bounds_without_endcaps(GetLocalBounds());
   bounds_without_endcaps.Inset(gfx::Insets::VH(0, border_radius));
   return bounds_without_endcaps;
--- a/chrome/browser/ui/views/location_bar/location_icon_view.cc
+++ b/chrome/browser/ui/views/location_bar/location_icon_view.cc
@@ -114,7 +114,7 @@ SkColor LocationIconView::GetForegroundC
 }
 
 bool LocationIconView::ShouldShowSeparator() const {
-  return false;
+  return ShouldShowLabel();
 }
 
 bool LocationIconView::ShouldShowLabelAfterAnimation() const {
@@ -409,7 +409,8 @@ gfx::Size LocationIconView::GetMinimumSi
 }
 
 void LocationIconView::SetBackgroundColor(SkColor color) {
-  SetBackground(views::CreateRoundedRectBackground(color, GetCornerRadii()));
+  SetBackground(views::CreateRoundedRectBackground(SK_ColorTRANSPARENT,
+      GetCornerRadii()));
 }
 
 BEGIN_METADATA(LocationIconView)
--- a/chrome/browser/ui/views/location_bar/location_icon_state_helper.cc
+++ b/chrome/browser/ui/views/location_bar/location_icon_state_helper.cc
@@ -69,16 +69,6 @@ bool ShouldShowSecurityChipText(const Lo
     return false;
   }
 
-  const GURL& url = model->GetURL();
-  if (url.SchemeIs(content::kChromeUIScheme) ||
-      url.SchemeIs(extensions::kExtensionScheme) ||
-      url.SchemeIs(url::kFileScheme) ||
-      url.SchemeIs(dom_distiller::kDomDistillerScheme) ||
-      (model->IsContextualTasksPage() &&
-       contextual_tasks::ShouldShowExpandedSecurityChip())) {
-    return true;
-  }
-
   return !model->GetSecureDisplayText().empty();
 }
 
