--- a/chrome/browser/ui/views/web_apps/frame_toolbar/web_app_frame_toolbar_utils.h
+++ b/chrome/browser/ui/views/web_apps/frame_toolbar/web_app_frame_toolbar_utils.h
@@ -11,7 +11,7 @@ class ToolbarButton;
 class ToolbarButtonProvider;
 
 #if BUILDFLAG(IS_MAC)
-inline constexpr int kWebAppMenuMargin = 7;
+inline constexpr int kWebAppMenuMargin = 4;
 #endif
 
 // Makes adjustments to |toolbar_button| for display in a web app frame.
--- a/chrome/browser/ui/views/web_apps/frame_toolbar/web_app_toolbar_button_container.cc
+++ b/chrome/browser/ui/views/web_apps/frame_toolbar/web_app_toolbar_button_container.cc
@@ -85,7 +85,7 @@ WebAppToolbarButtonContainer::WebAppTool
       .SetDefault(
           views::kMarginsKey,
           gfx::Insets::VH(
-              0, HorizontalPaddingBetweenPageActionsAndAppMenuButtons()))
+              0, GetLayoutConstant(LayoutConstant::kHeliumBasePadding) * 2))
       .SetCollapseMargins(true)
       .SetIgnoreDefaultMainAxisMargins(true)
       .SetCrossAxisAlignment(views::LayoutAlignment::kCenter)
--- a/chrome/browser/ui/views/location_bar/custom_tab_bar_view.cc
+++ b/chrome/browser/ui/views/location_bar/custom_tab_bar_view.cc
@@ -117,7 +117,7 @@ class CustomTabBarTitleOriginView : publ
                                  views::MaximumFlexSizeRule::kPreferred));
     location_label_ = AddChildView(std::move(location_label));
 
-    if (should_show_title) {
+    if ((false)) {
       auto title_label = std::make_unique<views::Label>(
           std::u16string(), views::style::CONTEXT_LABEL);
 
--- a/chrome/browser/ui/views/web_apps/frame_toolbar/web_app_frame_toolbar_utils.cc
+++ b/chrome/browser/ui/views/web_apps/frame_toolbar/web_app_frame_toolbar_utils.cc
@@ -25,7 +25,7 @@ void ConfigureWebAppToolbarButton(
 
 int WebAppFrameRightMargin() {
 #if BUILDFLAG(IS_MAC)
-  return kWebAppMenuMargin;
+  return 10;
 #else
   return HorizontalPaddingBetweenPageActionsAndAppMenuButtons();
 #endif
--- a/chrome/browser/ui/views/web_apps/frame_toolbar/web_app_navigation_button_container.cc
+++ b/chrome/browser/ui/views/web_apps/frame_toolbar/web_app_navigation_button_container.cc
@@ -31,12 +31,7 @@
 namespace {
 
 constexpr int kPaddingBetweenNavigationButtons = 5;
-
-#if BUILDFLAG(IS_CHROMEOS)
 constexpr int kWebAppFrameLeftMargin = 2;
-#else
-constexpr int kWebAppFrameLeftMargin = 7;
-#endif
 
 }  // namespace
 
