--- a/chrome/browser/ui/views/location_bar/location_bar_view.cc
+++ b/chrome/browser/ui/views/location_bar/location_bar_view.cc
@@ -517,7 +517,6 @@ void LocationBarView::Init() {
     }
     params.types_enabled.push_back(PageActionIconType::kPwaInstall);
     params.types_enabled.push_back(PageActionIconType::kTranslate);
-    params.types_enabled.push_back(PageActionIconType::kZoom);
     params.types_enabled.push_back(PageActionIconType::kFileSystemAccess);
 
     params.types_enabled.push_back(PageActionIconType::kMemorySaver);
--- a/chrome/browser/ui/web_applications/app_browser_controller.cc
+++ b/chrome/browser/ui/web_applications/app_browser_controller.cc
@@ -435,7 +435,6 @@ AppBrowserController::GetTitleBarPageAct
 
   std::vector<PageActionIconType> types_enabled;
   types_enabled.push_back(PageActionIconType::kFind);
-  types_enabled.push_back(PageActionIconType::kZoom);
   types_enabled.push_back(PageActionIconType::kFileSystemAccess);
 
   return types_enabled;
--- a/chrome/browser/ui/views/zoom/zoom_view_controller.cc
+++ b/chrome/browser/ui/views/zoom/zoom_view_controller.cc
@@ -90,16 +90,7 @@ void ZoomViewController::UpdatePageActio
 
 void ZoomViewController::UpdatePageActionIconVisibility(
     bool is_bubble_visible) {
-  zoom::ZoomController* zoom_controller =
-      zoom::ZoomController::FromWebContents(GetWebContents());
-  CHECK(zoom_controller);
-
-  const bool is_at_default_zoom = zoom_controller->IsAtDefaultZoom();
-  if (is_at_default_zoom && !is_bubble_visible) {
-    page_action_controller_->Hide(kActionZoomNormal);
-  } else {
-    page_action_controller_->Show(kActionZoomNormal);
-  }
+  page_action_controller_->Hide(kActionZoomNormal);
 }
 
 void ZoomViewController::UpdateBubbleVisibility(bool prefer_to_show_bubble,
@@ -143,8 +134,7 @@ bool ZoomViewController::CanBubbleBeVisi
   // If neither of these is true, we only show the bubble if the zoom level
   // is not at the default (`!is_zoom_at_default`), indicating that zoom is
   // active.
-  return (prefer_to_show_bubble || IsBubbleVisible()) ? true
-                                                      : !is_zoom_at_default;
+  return false;
 }
 
 content::WebContents* ZoomViewController::GetWebContents() const {
