--- a/chrome/browser/ui/views/profiles/profile_menu_view.cc
+++ b/chrome/browser/ui/views/profiles/profile_menu_view.cc
@@ -1045,6 +1046,7 @@ void ProfileMenuView::MaybeBuildGoogleSe
 }
 
 void ProfileMenuView::MaybeBuildManageGoogleAccountButton() {
+  return;
   CHECK(!profile().IsGuestSession());
   signin::IdentityManager* identity_manager =
       IdentityManagerFactory::GetForProfile(&profile());
