--- a/components/password_manager/core/browser/password_suggestion_generator.cc
+++ b/components/password_manager/core/browser/password_suggestion_generator.cc
@@ -397,6 +397,10 @@ PasswordSuggestionGenerator::PasswordSug
 
 void PasswordSuggestionGenerator::AppendOptionalFooterSection(
     std::vector<autofill::Suggestion>* suggestions) const {
+  if ((true)) {
+    return;
+  }
+
   bool has_webauthn_credential = std::ranges::any_of(
       *suggestions,
       [](SuggestionType type) {
