--- a/chrome/browser/ui/views/location_bar/location_bar_view.cc
+++ b/chrome/browser/ui/views/location_bar/location_bar_view.cc
@@ -513,4 +513,5 @@ void LocationBarView::Init() {
     params.types_enabled.push_back(PageActionIconType::kClickToCall);
+    params.types_enabled.push_back(PageActionIconType::kManagePasswords);
     if (!apps::features::ShouldShowLinkCapturingUX()) {
       params.types_enabled.push_back(PageActionIconType::kIntentPicker);
     }
--- a/chrome/browser/ui/web_applications/app_browser_controller.cc
+++ b/chrome/browser/ui/web_applications/app_browser_controller.cc
@@ -411,4 +411,5 @@ std::vector<actions::ActionId> AppBrowse
       kActionFind,
+      kActionShowPasswordsBubbleOrPage,
       kActionZoomNormal,
       kActionShowFileSystemAccess,
       kActionShowCookieControls,
@@ -436,2 +437,3 @@ AppBrowserController::GetTitleBarPageAct
   types_enabled.push_back(PageActionIconType::kFind);
+  types_enabled.push_back(PageActionIconType::kManagePasswords);
   types_enabled.push_back(PageActionIconType::kFileSystemAccess);
