--- a/chrome/browser/ui/views/location_bar/location_bar_view.cc
+++ b/chrome/browser/ui/views/location_bar/location_bar_view.cc
@@ -507,6 +507,7 @@ void LocationBarView::Init() {
       params.types_enabled.push_back(PageActionIconType::kOptimizationGuide);
     }
     params.types_enabled.push_back(PageActionIconType::kClickToCall);
+    params.types_enabled.push_back(PageActionIconType::kManagePasswords);
     if (!apps::features::ShouldShowLinkCapturingUX()) {
       params.types_enabled.push_back(PageActionIconType::kIntentPicker);
     }
--- a/chrome/browser/ui/web_applications/app_browser_controller.cc
+++ b/chrome/browser/ui/web_applications/app_browser_controller.cc
@@ -409,6 +409,7 @@ std::vector<actions::ActionId> AppBrowse
 
   std::vector<actions::ActionId> types_enabled = {
       kActionFind,
+      kActionShowPasswordsBubbleOrPage,
       kActionZoomNormal,
       kActionShowFileSystemAccess,
       kActionShowCookieControls,
@@ -434,6 +435,7 @@ AppBrowserController::GetTitleBarPageAct
 
   std::vector<PageActionIconType> types_enabled;
   types_enabled.push_back(PageActionIconType::kFind);
+  types_enabled.push_back(PageActionIconType::kManagePasswords);
   types_enabled.push_back(PageActionIconType::kZoom);
   types_enabled.push_back(PageActionIconType::kFileSystemAccess);
 
