--- a/chrome/browser/ui/browser_actions.cc
+++ b/chrome/browser/ui/browser_actions.cc
@@ -1295,7 +1295,40 @@ void BrowserActions::InitializeChromeMen
             .SetVisible(ShouldShowChromeLabsUI(profile))
             .Build());
   }
 
+  root_action_item_->AddChild(
+      ChromeMenuAction(
+          base::BindRepeating(
+              [](BrowserWindowInterface* bwi, actions::ActionItem* item,
+                 actions::ActionInvocationContext context) {
+                TabStripModel* const tab_strip_model =
+                    bwi->GetTabStripModel();
+                content::WebContents* const web_contents =
+                    tab_strip_model->GetActiveWebContents();
+                if (PasswordsModelDelegateFromWebContents(web_contents)
+                        ->GetState() == password_manager::ui::INACTIVE_STATE) {
+                  chrome::ShowPasswordManager(bwi);
+                } else {
+                  auto* const controller =
+                      ManagePasswordsUIController::FromWebContents(
+                          web_contents);
+                  if (controller->IsShowingBubble()) {
+                    controller->HideBubble(
+                        /*initiated_by_bubble_manager=*/false);
+                  } else {
+                    chrome::ManagePasswordsForPage(bwi);
+                  }
+                }
+              },
+              bwi),
+          kActionShowPasswordsBubbleOrPage, IDS_VIEW_PASSWORDS,
+          IDS_VIEW_PASSWORDS,
+          features::IsRoundedIconsEnabled()
+              ? vector_icons::kPasswordManagerIcon
+              : vector_icons::kPasswordManagerOldIcon)
+          .SetEnabled(!is_guest_session)
+          .Build());
+
   root_action_item_->AddChild(
       actions::ActionItem::Builder(
           base::BindRepeating([](actions::ActionItem* item,
--- a/chrome/browser/ui/web_applications/app_browser_controller.cc
+++ b/chrome/browser/ui/web_applications/app_browser_controller.cc
@@ -436,6 +436,7 @@ std::vector<actions::ActionId> AppBrowse
 
   std::vector<actions::ActionId> types_enabled = {
       kActionFind,
+      kActionShowPasswordsBubbleOrPage,
       kActionShowZoomBubble,
       kActionShowFileSystemAccess,
   };
@@ -459,6 +460,7 @@ AppBrowserController::GetTitleBarPageAct
 
   std::vector<PageActionIconType> types_enabled;
   types_enabled.push_back(PageActionIconType::kFind);
+  types_enabled.push_back(PageActionIconType::kManagePasswords);
   types_enabled.push_back(PageActionIconType::kFileSystemAccess);
 
   return types_enabled;
