--- a/chrome/browser/ui/views/location_bar/selected_keyword_view.cc
+++ b/chrome/browser/ui/views/location_bar/selected_keyword_view.cc
@@ -54,8 +54,7 @@ SelectedKeywordView::GetKeywordLabelName
     return names;
   }
 
-  names.full_name =
-      l10n_util::GetStringFUTF16(IDS_OMNIBOX_KEYWORD_TEXT_MD, names.short_name);
+  names.full_name = names.short_name;
   return names;
 }
 
@@ -70,6 +69,8 @@ SelectedKeywordView::SelectedKeywordView
   partial_label_.SetVisible(false);
   label()->SetElideBehavior(gfx::FADE_TAIL);
 
+  SetExpandedLabelAdditionalInsets(views::Inset1D(2, 0));
+
   // TODO(crbug.com/40890218): `IconLabelBubbleView::GetAccessibleNodeData`
   // would set the name to explicitly empty when the name was missing.
   // That function no longer exists. As a result we need to handle that here.
@@ -193,7 +194,7 @@ const std::u16string& SelectedKeywordVie
 
 int SelectedKeywordView::GetExtraInternalSpacing() const {
   // Align the label text with the suggestion text.
-  return 14;
+  return 2;
 }
 
 void SelectedKeywordView::SetLabelForCurrentWidth() {
--- a/components/search_engines/template_url.cc
+++ b/components/search_engines/template_url.cc
@@ -2393,11 +2393,7 @@ std::u16string TemplateURL::GetFullName(
   if (is_ask_type()) {
     return l10n_util::GetStringFUTF16(IDS_OMNIBOX_SELECTED_KEYWORD_ASK_TEXT,
                                       short_name);
-  } else if (starter_pack_id() ==
-             template_url_starter_pack_data::StarterPackId::kPage) {
-    return l10n_util::GetStringUTF16(IDS_STARTER_PACK_PAGE_KEYWORD_TEXT);
-  } else if (type() == TemplateURL::OMNIBOX_API_EXTENSION) {
-    return short_name;
   }
-  return l10n_util::GetStringFUTF16(IDS_OMNIBOX_KEYWORD_TEXT_MD, short_name);
+
+  return short_name;
 }
