--- a/chrome/browser/ui/views/side_panel/side_panel_resize_area.cc
+++ b/chrome/browser/ui/views/side_panel/side_panel_resize_area.cc
@@ -20,6 +20,7 @@
 #include "ui/views/controls/focus_ring.h"
 #include "ui/views/controls/image_view.h"
 #include "ui/views/controls/resize_area.h"
+#include "ui/views/controls/separator.h"
 #include "ui/views/layout/flex_layout.h"
 #include "ui/views/view_class_properties.h"
 
@@ -29,7 +30,7 @@ SidePanelResizeHandle::SidePanelResizeHa
     : side_panel_(side_panel) {
   SetProperty(views::kElementIdentifierKey, kSidePanelResizeHandleElementId);
   SetVisible(false);
-  gfx::Size preferred_size(4, 24);
+  gfx::Size preferred_size(2, 24);
   SetPreferredSize(preferred_size);
   SetCanProcessEventsWithinSubtree(false);
 
@@ -142,19 +143,30 @@ void SidePanelResizeArea::OnBlur() {
 
 void SidePanelResizeArea::Layout(PassKey) {
   LayoutSuperclass<ResizeArea>(this);
-  // The side panel resize area should draw on top of its parent's border.
+  // The side panel resize area should fill the margin outside the content
+  // outline without covering the outline itself.
   gfx::Rect local_bounds = parent()->GetLocalBounds();
   gfx::Rect contents_bounds = parent()->GetContentsBounds();
 
   gfx::Rect resize_bounds;
   if ((side_panel_->IsRightAligned() && !base::i18n::IsRTL()) ||
       (!side_panel_->IsRightAligned() && base::i18n::IsRTL())) {
+    int resize_width = contents_bounds.x() - views::Separator::kThickness;
+    if (resize_width < 0) {
+      resize_width = 0;
+    }
     resize_bounds = gfx::Rect(local_bounds.x(), local_bounds.y(),
-                              contents_bounds.x(), local_bounds.height());
+                              resize_width, local_bounds.height());
   } else {
-    resize_bounds = gfx::Rect(contents_bounds.right(), local_bounds.y(),
-                              local_bounds.right() - contents_bounds.right(),
-                              local_bounds.height());
+    const int resize_x =
+        contents_bounds.right() + views::Separator::kThickness;
+    int resize_width = local_bounds.right() - resize_x;
+    if (resize_width < 0) {
+      resize_width = 0;
+    }
+    resize_bounds =
+        gfx::Rect(resize_x, local_bounds.y(), resize_width,
+                  local_bounds.height());
   }
 
   SetBoundsRect(resize_bounds);
--- a/chrome/browser/ui/views/side_panel/side_panel.cc
+++ b/chrome/browser/ui/views/side_panel/side_panel.cc
@@ -27,6 +27,7 @@
 #include "chrome/browser/ui/views/chrome_layout_provider.h"
 #include "chrome/browser/ui/views/frame/browser_view.h"
 #include "chrome/browser/ui/views/frame/themed_background.h"
+#include "chrome/browser/ui/views/helium/frame_corner_radius.h"
 #include "chrome/browser/ui/views/side_panel/side_panel_animation_perf_reporter.h"
 #include "chrome/browser/ui/views/side_panel/side_panel_resize_area.h"
 #include "chrome/common/pref_names.h"
@@ -60,9 +61,13 @@
 #include "ui/views/layout/layout_provider.h"
 #include "ui/views/view_class_properties.h"
 #include "ui/views/view_observer.h"
+#include "ui/views/widget/widget.h"
 
 namespace {
 
+constexpr int kContentMargin = 3;
+constexpr int kResizeAreaMargin = 6;
+
 // Converts from animation motion to animation type.
 SidePanelAnimationType AnimationMotionToType(BrowserAnimationMotion motion) {
   if (motion == SidePanelAnimations::kOpen) {
@@ -79,23 +84,33 @@ SidePanelAnimationType AnimationMotionTo
 
 // This thickness includes the solid-color background and the inner round-rect
 // border-color stroke. It does not include the outer-color separator.
-int GetBorderThickness() {
-  return 8 + views::Separator::kThickness;
+int GetBorderThickness(int margin) {
+  return margin + views::Separator::kThickness;
 }
 
 // This is how many units of the toolbar are essentially expected to be
 // background.
 constexpr int kOverlapFromToolbar = 4;
 
-// We want the border to visually look like GetBorderThickness() units on all
-// sides except the top. On the top side, background is drawn on top of the
-// top-content separator and some units of background inside the toolbar (or
-// bookmarks bar) itself. Subtract both of those to not get visually-excessive
-// padding.
-gfx::Insets GetBorderInsets() {
-  int border_thickness = GetBorderThickness();
-  return gfx::Insets::TLBR(-kOverlapFromToolbar, border_thickness,
-                           border_thickness, border_thickness);
+bool IsResizeAreaOnLeft(const SidePanel* side_panel) {
+  return (side_panel->IsRightAligned() && !base::i18n::IsRTL()) ||
+         (!side_panel->IsRightAligned() && base::i18n::IsRTL());
+}
+
+// We want the border to visually look like kContentMargin units on all sides
+// except the top and the resize area edge. On the top side, background is drawn
+// on top of the top-content separator and some units of background inside the
+// toolbar (or bookmarks bar) itself. Subtract both of those to not get
+// visually-excessive padding.
+gfx::Insets GetBorderInsets(const SidePanel* side_panel) {
+  const int content_thickness = GetBorderThickness(kContentMargin);
+  const int resize_area_thickness = GetBorderThickness(kResizeAreaMargin);
+  const bool resize_area_on_left = IsResizeAreaOnLeft(side_panel);
+  return gfx::Insets::TLBR(
+      -kOverlapFromToolbar,
+      resize_area_on_left ? resize_area_thickness : content_thickness,
+      content_thickness,
+      resize_area_on_left ? content_thickness : resize_area_thickness);
 }
 
 SidePanel::HorizontalAlignment GetHorizontalAlignment(
@@ -110,14 +125,52 @@ SidePanel::HorizontalAlignment GetHorizo
                           : SidePanel::HorizontalAlignment::kLeft;
 }
 
+gfx::RoundedCornersF GetContentRoundedCorners(
+    const BrowserView* browser_view,
+    const SidePanel* side_panel) {
+  const int generic_radius = GetGenericFrameRadius();
+  const bool is_fullscreen =
+      side_panel->GetWidget() && side_panel->GetWidget()->IsFullscreen();
+  const int window_edge_radius =
+      is_fullscreen ? generic_radius : GetPaddedFrameRadius();
+  const bool lower_corner_touches_window_edge =
+      !browser_view->ShouldDrawVerticalTabStrip() ||
+      side_panel->IsRightAligned() != base::i18n::IsRTL();
+  const int lower_window_edge_radius =
+      lower_corner_touches_window_edge ? window_edge_radius : generic_radius;
+
+  return side_panel->IsRightAligned()
+             ? gfx::RoundedCornersF(generic_radius, generic_radius,
+                                    lower_window_edge_radius, generic_radius)
+             : gfx::RoundedCornersF(generic_radius, generic_radius,
+                                    generic_radius, lower_window_edge_radius);
+}
+
+void RoundedCornersToSkVectors(const gfx::RoundedCornersF& radii,
+                               SkVector sk_radii[4],
+                               float scale = 1.0f) {
+  const auto get_radius_vector = [scale](float radius) {
+    const float scaled_radius = radius * scale;
+    return SkVector{scaled_radius, scaled_radius};
+  };
+  sk_radii[SkRRect::kUpperLeft_Corner] =
+      get_radius_vector(radii.upper_left());
+  sk_radii[SkRRect::kUpperRight_Corner] =
+      get_radius_vector(radii.upper_right());
+  sk_radii[SkRRect::kLowerRight_Corner] =
+      get_radius_vector(radii.lower_right());
+  sk_radii[SkRRect::kLowerLeft_Corner] =
+      get_radius_vector(radii.lower_left());
+}
+
 // This border paints the toolbar color around the side panel content and draws
 // a roundrect viewport around the side panel content. The border can have
 // rounded corners of its own.
 class SidePanelBorder : public views::Border {
  public:
-  explicit SidePanelBorder(BrowserView* browser_view)
-      : browser_view_(browser_view) {
-    SetColor(kColorSidePanelContentAreaSeparator);
+  SidePanelBorder(SidePanel* side_panel, BrowserView* browser_view)
+      : side_panel_(side_panel), browser_view_(browser_view) {
+    SetColor(kColorToolbarContentAreaSeparator);
   }
 
   SidePanelBorder(const SidePanelBorder&) = delete;
@@ -142,10 +195,6 @@ class SidePanelBorder : public views::Bo
         view.GetLocalBounds(), view.layer()->device_scale_factor());
 
     gfx::RectF scaled_contents_bounds_f = scaled_view_bounds_f;
-    const float corner_radius =
-        dsf * view.GetLayoutProvider()->GetDistanceMetric(
-                  ChromeDistanceMetric::
-                      DISTANCE_CONTENT_HEIGHT_SIDE_PANEL_CONTENT_RADIUS);
     const gfx::InsetsF insets_in_pixels(
         gfx::ConvertInsetsToPixels(GetInsets(), dsf));
     scaled_contents_bounds_f.Inset(insets_in_pixels);
@@ -153,8 +202,12 @@ class SidePanelBorder : public views::Bo
     // Use ToEnclosedRect to make sure that the clip bounds never end up larger
     // than the child view.
     gfx::Rect clip_bounds = ToEnclosedRect(scaled_contents_bounds_f);
-    SkRRect rect = SkRRect::MakeRectXY(gfx::RectToSkRect(clip_bounds),
-                                       corner_radius, corner_radius);
+    SkVector content_radii[4];
+    RoundedCornersToSkVectors(GetContentRoundedCorners(browser_view_,
+                                                       side_panel_),
+                              content_radii, dsf);
+    SkRRect rect = SkRRect::MakeRectRadii(gfx::RectToSkRect(clip_bounds),
+                                          content_radii);
 
     // Clip out the content area from the background about to be painted.
     canvas->sk_canvas()->clipRRect(rect, SkClipOp::kDifference,
@@ -167,11 +220,8 @@ class SidePanelBorder : public views::Bo
       gfx::ScopedCanvas scoped_rescale(canvas);
       canvas->Scale(dsf, dsf);
 
-      const SkVector border_radii[4] = {
-          {border_radii_.upper_left(), border_radii_.upper_left()},
-          {border_radii_.upper_right(), border_radii_.upper_right()},
-          {border_radii_.lower_right(), border_radii_.lower_right()},
-          {border_radii_.lower_left(), border_radii_.lower_left()}};
+      SkVector border_radii[4];
+      RoundedCornersToSkVectors(border_radii_, border_radii);
 
       const SkPath rounded_border_path = SkPath::RRect(SkRRect::MakeRectRadii(
           gfx::RectToSkRect(view.GetLocalBounds()), border_radii));
@@ -219,9 +269,10 @@ class SidePanelBorder : public views::Bo
     // inset is outside the SidePanel itself, but not outside the BorderView. If
     // there is a header we want to increase the top inset to give room for the
     // header to paint on top of the border area.
+    const gfx::Insets border_insets = GetBorderInsets(side_panel_);
     int top_inset =
-        views::Separator::kThickness + header_height_ - GetBorderInsets().top();
-    return GetBorderInsets() + gfx::Insets::TLBR(top_inset, 0, 0, 0);
+        views::Separator::kThickness + header_height_ - border_insets.top();
+    return border_insets + gfx::Insets::TLBR(top_inset, 0, 0, 0);
   }
   gfx::Size GetMinimumSize() const override {
     return gfx::Size(GetInsets().width(), GetInsets().height());
@@ -231,6 +282,7 @@ class SidePanelBorder : public views::Bo
   int header_height_ = 0;
   gfx::RoundedCornersF border_radii_;
   bool outline_visible_ = true;
+  const raw_ptr<SidePanel> side_panel_;
   const raw_ptr<BrowserView> browser_view_;
 };
 
@@ -247,10 +299,8 @@ class ContentParentBackground : public v
 
   void Paint(gfx::Canvas* canvas, views::View* view) const override {
     gfx::RoundedCornersF radii = get_rounded_corners_callback_.Run();
-    SkVector sk_radii[4] = {{radii.upper_left(), radii.upper_left()},
-                            {radii.upper_right(), radii.upper_right()},
-                            {radii.lower_right(), radii.lower_right()},
-                            {radii.lower_left(), radii.lower_left()}};
+    SkVector sk_radii[4];
+    RoundedCornersToSkVectors(radii, sk_radii);
     SkRRect rrect;
     rrect.setRectRadii(gfx::RectToSkRect(view->GetLocalBounds()), sk_radii);
     SkPath path = SkPath::RRect(rrect);
@@ -276,10 +326,12 @@ class ContentParentView : public views::
   METADATA_HEADER(ContentParentView, views::View)
 
  public:
-  explicit ContentParentView(BrowserView* browser_view,
+  explicit ContentParentView(SidePanel* side_panel,
+                             BrowserView* browser_view,
                              bool should_round_corners,
                              SidePanelEntry::PanelType type)
-      : browser_view_(browser_view),
+      : side_panel_(side_panel),
+        browser_view_(browser_view),
         should_round_corners_(should_round_corners),
         type_(type) {
     SetUseDefaultFillLayout(true);
@@ -296,6 +348,17 @@ class ContentParentView : public views::
 
   ~ContentParentView() override = default;
 
+  void UpdateRoundedCorners() {
+    if (!should_round_corners_) {
+      return;
+    }
+    const gfx::RoundedCornersF radii = GetRoundedCorners();
+    for (views::View* child : children()) {
+      UpdateChildRoundedCorners(child, radii);
+    }
+    SchedulePaint();
+  }
+
  private:
   void AddedToWidget() override {
     SetBackground(std::make_unique<ContentParentBackground>(
@@ -313,22 +376,7 @@ class ContentParentView : public views::
     // corners initially, set them in OnChildViewAdded, which is called after
     // View::ViewHierarchyChanged.
     // If the child is a WebView or paints to a layer, round its corners.
-    if (views::IsViewClass<views::WebView>(child)) {
-      views::AsViewClass<views::WebView>(child)->holder()->SetCornerRadii(
-          GetRoundedCorners());
-    }
-    // Try to detect if the child is a views::View wrapper of a WebView. If so,
-    // round its corners.
-    if (child->children().size() == 1 &&
-        views::IsViewClass<views::WebView>(child->children()[0])) {
-      views::AsViewClass<views::WebView>(child->children()[0])
-          ->holder()
-          ->SetCornerRadii(GetRoundedCorners());
-    }
-    if (child->layer()) {
-      child->layer()->SetIsFastRoundedCorner(true);
-      child->layer()->SetRoundedCornerRadius(GetRoundedCorners());
-    }
+    UpdateChildRoundedCorners(child, GetRoundedCorners());
   }
 
   void OnWebContentsAttached(views::WebView* web_view) {
@@ -361,18 +409,32 @@ class ContentParentView : public views::
   }
 
   gfx::RoundedCornersF GetRoundedCorners() {
-    ChromeDistanceMetric corner_radius =
-        type_ == SidePanelEntry::PanelType::kToolbar
-            ? ChromeDistanceMetric::
-                  DISTANCE_TOOLBAR_HEIGHT_SIDE_PANEL_CONTENT_RADIUS
-            : ChromeDistanceMetric::
-                  DISTANCE_CONTENT_HEIGHT_SIDE_PANEL_CONTENT_RADIUS;
-    return should_round_corners_ && GetLayoutProvider()
-               ? gfx::RoundedCornersF(
-                     GetLayoutProvider()->GetDistanceMetric(corner_radius))
+    return should_round_corners_
+               ? GetContentRoundedCorners(browser_view_, side_panel_)
                : gfx::RoundedCornersF();
   }
 
+  void UpdateChildRoundedCorners(views::View* child,
+                                 const gfx::RoundedCornersF& radii) {
+    if (views::IsViewClass<views::WebView>(child)) {
+      views::AsViewClass<views::WebView>(child)->holder()->SetCornerRadii(
+          radii);
+    }
+    // Try to detect if the child is a views::View wrapper of a WebView. If so,
+    // round its corners.
+    if (child->children().size() == 1 &&
+        views::IsViewClass<views::WebView>(child->children()[0])) {
+      views::AsViewClass<views::WebView>(child->children()[0])
+          ->holder()
+          ->SetCornerRadii(radii);
+    }
+    if (child->layer()) {
+      child->layer()->SetIsFastRoundedCorner(true);
+      child->layer()->SetRoundedCornerRadius(radii);
+    }
+  }
+
+  const raw_ptr<SidePanel> side_panel_;
   const raw_ptr<BrowserView> browser_view_;
   bool should_round_corners_ = false;
   SidePanelEntry::PanelType type_;
@@ -394,9 +456,9 @@ class SidePanel::BorderView : public vie
   METADATA_HEADER(BorderView, views::View)
 
  public:
-  explicit BorderView(BrowserView* browser_view) {
+  BorderView(SidePanel* side_panel, BrowserView* browser_view) {
     SetVisible(false);
-    auto border = std::make_unique<SidePanelBorder>(browser_view);
+    auto border = std::make_unique<SidePanelBorder>(side_panel, browser_view);
     border_ = border.get();
     SetBorder(std::move(border));
     // Don't allow the view to process events. If we do allow this then events
@@ -504,12 +566,12 @@ SidePanel::SidePanel(BrowserView* browse
   // parent view. content_parent_view_ is added first so it exists behind
   // border_view_ and resize_area_.
   content_parent_view_ = AddChildView(std::make_unique<ContentParentView>(
-      browser_view, /*should_round_corners=*/!has_border, type));
+      this, browser_view, /*should_round_corners=*/!has_border, type));
   content_parent_view_->SetVisible(false);
 
   if (has_border) {
     std::unique_ptr<BorderView> border_view =
-        std::make_unique<BorderView>(browser_view);
+        std::make_unique<BorderView>(this, browser_view);
     border_view_ = border_view.get();
     AddChildView(std::move(border_view));
   }
@@ -541,9 +603,7 @@ SidePanel::SidePanel(BrowserView* browse
   // default.
   SetPanelWidth(GetMinimumSize().width());
 
-  if (has_border) {
-    SetBorder(views::CreateEmptyBorder(GetBorderInsets()));
-  }
+  UpdateBorderInsets();
 
   SetProperty(views::kElementIdentifierKey, kSidePanelElementId);
 }
@@ -581,6 +641,29 @@ void SidePanel::SetBackgroundRadii(const
   }
 }
 
+void SidePanel::UpdateBorderInsets() {
+  if (!border_view_ && !header_view_) {
+    SetBorder(nullptr);
+    return;
+  }
+
+  gfx::Insets border_insets = GetBorderInsets(this);
+  if (header_view_) {
+    int top_inset = header_view_->height() - border_insets.top();
+    border_insets += gfx::Insets::TLBR(top_inset, 0, 0, 0);
+  }
+
+  SetBorder(views::CreateEmptyBorder(border_insets));
+}
+
+void SidePanel::UpdateRoundedCorners() {
+  if (border_view_) {
+    border_view_->SchedulePaint();
+  }
+  static_cast<ContentParentView*>(content_parent_view_.get())
+      ->UpdateRoundedCorners();
+}
+
 void SidePanel::UpdateWidthOnEntryChanged() {
   SidePanelUI* side_panel_ui =
       browser_view_->browser()->GetFeatures().side_panel_ui();
@@ -609,7 +692,7 @@ void SidePanel::UpdateWidthOnEntryChange
     SetPanelWidth(width_from_pref.value());
   } else {
     SetPanelWidth(side_panel_ui->GetCurrentEntryDefaultContentWidth(type_) +
-                  GetBorderInsets().width());
+                  GetBorderInsets(this).width());
   }
 }
 
@@ -620,7 +703,8 @@ bool SidePanel::IsRightAligned() const {
 gfx::Size SidePanel::GetMinimumSize() const {
   const int min_height = 0;
   return gfx::Size(
-      SidePanelEntry::kSidePanelDefaultContentWidth + GetBorderInsets().width(),
+      SidePanelEntry::kSidePanelDefaultContentWidth +
+          GetBorderInsets(this).width(),
       min_height);
 }
 
@@ -680,13 +764,10 @@ void SidePanel::AddHeaderView(std::uniqu
   }
   // Update the border so that the insets include space for the header to be
   // placed on top of the border area.
-  int top_inset = header_view_->height() - GetBorderInsets().top();
-  SetBorder(views::CreateEmptyBorder(GetBorderInsets() +
-                                     gfx::Insets::TLBR(top_inset, 0, 0, 0)));
+  UpdateBorderInsets();
 }
 
 void SidePanel::RemoveHeaderView() {
-  SetBorder(views::CreateEmptyBorder(GetBorderInsets().set_top(0)));
   if (border_view_) {
     border_view_->HeaderViewChanged(nullptr);
   }
@@ -694,6 +775,7 @@ void SidePanel::RemoveHeaderView() {
     auto header_view = RemoveChildViewT(header_view_);
     header_view_ = nullptr;
   }
+  UpdateBorderInsets();
 }
 
 void SidePanel::SetOutlineVisibility(bool visible) {
@@ -708,11 +790,14 @@ gfx::Size SidePanel::GetContentSizeUpper
   const int side_panel_height =
       height() > 0 ? height() : browser_view_->height();
 
-  return gfx::Size(std::max(0, side_panel_width - GetBorderInsets().width()),
-                   std::max(0, side_panel_height - GetBorderInsets().height()));
+  return gfx::Size(
+      std::max(0, side_panel_width - GetBorderInsets(this).width()),
+      std::max(0, side_panel_height - GetBorderInsets(this).height()));
 }
 
 void SidePanel::OnBoundsChanged(const gfx::Rect& previous_bounds) {
+  UpdateRoundedCorners();
+
   if (previous_bounds.width() != width() && keyboard_resized_) {
     keyboard_resized_ = false;
     AnnounceResize();
@@ -844,7 +929,7 @@ void SidePanel::RecordMetricsIfResized()
       return;
     }
 
-    int side_panel_contents_width = width() - GetBorderInsets().width();
+    int side_panel_contents_width = width() - GetBorderInsets(this).width();
     int browser_window_width = browser_view_->width();
     SidePanelMetrics::RecordSidePanelResizeMetrics(
         type_, id.value(), side_panel_contents_width, browser_window_width);
@@ -921,9 +1006,7 @@ void SidePanel::UpdateVisibility(bool sh
       border_view_->layer()->SetFillsBoundsOpaquely(false);
       if (header_view_ && header_view_->GetVisible()) {
         border_view_->HeaderViewChanged(header_view_);
-        int top_inset = header_view_->height() - GetBorderInsets().top();
-        SetBorder(views::CreateEmptyBorder(
-            GetBorderInsets() + gfx::Insets::TLBR(top_inset, 0, 0, 0)));
+        UpdateBorderInsets();
       }
     } else {
       border_view_->DestroyLayer();
@@ -1012,6 +1095,8 @@ void SidePanel::UpdateHorizontalAlignmen
       GetHorizontalAlignment(browser_view_->GetProfile()->GetPrefs(), type_,
                              use_default_horizontal_alignment_);
 
+  UpdateBorderInsets();
+  UpdateRoundedCorners();
   InvalidateLayout();
 }
 
--- a/chrome/browser/ui/views/side_panel/side_panel.h
+++ b/chrome/browser/ui/views/side_panel/side_panel.h
@@ -137,6 +137,8 @@ class SidePanel : public views::Accessib
   void AnnounceResize();
 
   void UpdateHorizontalAlignment();
+  void UpdateBorderInsets();
+  void UpdateRoundedCorners();
 
   // views::View:
   void OnBoundsChanged(const gfx::Rect& previous_bounds) override;
--- a/chrome/browser/ui/views/side_panel/side_panel_header.cc
+++ b/chrome/browser/ui/views/side_panel/side_panel_header.cc
@@ -23,7 +23,7 @@
 
 namespace {
 // The minimum cross axis size should the expected height of the header.
-constexpr int kDefaultSidePanelHeaderHeight = 40;
+constexpr int kDefaultSidePanelHeaderHeight = 32;
 }  // namespace
 
 SidePanelHeader::SidePanelHeader(
