--- a/chrome/browser/ui/layout_constants.cc
+++ b/chrome/browser/ui/layout_constants.cc
@@ -103,7 +103,7 @@ int GetLayoutConstant(LayoutConstant constant) {
     case LayoutConstant::kTabCloseButtonSize:
       return touch_ui ? 24 : (features::IsRoundedIconsEnabled() ? 14 : 16);
     case LayoutConstant::kTabHeight:
-      return 31;
+      return 36;
     case LayoutConstant::kTabStripHeight:
       return GetLayoutConstant(LayoutConstant::kTabHeight) +
              GetLayoutConstant(LayoutConstant::kTabStripPadding);
