--- a/chrome/browser/ui/views/chrome_layout_provider.cc
+++ b/chrome/browser/ui/views/chrome_layout_provider.cc
@@ -169,7 +169,7 @@ int ChromeLayoutProvider::GetDistanceMet
       // Therefore, the total infobar height is 32dp + 2 * 12.
       return base::FeatureList::IsEnabled(features::kInfobarRefresh)
                  ? 32 + 2 * 12
-                 : 36 + 2 * 8;
+                 : 36 + 2 * 3;
     case DISTANCE_PERMISSION_PROMPT_HORIZONTAL_ICON_LABEL_PADDING:
       return 8;
     case DISTANCE_RICH_HOVER_BUTTON_ICON_HORIZONTAL:
--- a/chrome/browser/ui/views/infobars/infobar_container_view.cc
+++ b/chrome/browser/ui/views/infobars/infobar_container_view.cc
@@ -122,8 +122,7 @@ void InfoBarContainerView::Layout(PassKe
   // there drawn by the shadow code (so we don't have to extend our bounds out
   // to be able to draw it; see comments in CalculatePreferredSize() on why the
   // shadow is drawn outside the container bounds).
-  content_shadow_->SetBounds(0, top, width(),
-                             content_shadow_->GetPreferredSize().height());
+  content_shadow_->SetBounds(0, top, width(), 1);
 }
 
 gfx::Size InfoBarContainerView::CalculatePreferredSize(
