--- a/chrome/browser/ui/webui/cr_components/history/history_util.cc
+++ b/chrome/browser/ui/webui/cr_components/history/history_util.cc
@@ -51,7 +51,7 @@ content::WebUIDataSource* HistoryUtil::P
       {"expandSessionButton", IDS_HISTORY_OTHER_SESSIONS_EXPAND_SESSION},
       {"foundSearchResults", IDS_HISTORY_FOUND_SEARCH_RESULTS},
       {"historyMenuButton", IDS_HISTORY_HISTORY_MENU_DESCRIPTION},
-      {"historyMenuItem", IDS_HISTORY_HISTORY_MENU_ITEM},
+      {"historyMenuItem", IDS_HISTORY_TITLE},
       {"itemsSelected", IDS_HISTORY_ITEMS_SELECTED},
       {"itemsUnselected", IDS_HISTORY_ITEMS_UNSELECTED},
       {"loading", IDS_HISTORY_LOADING},
--- a/chrome/browser/ui/webui/settings/settings_localized_strings_provider.cc
+++ b/chrome/browser/ui/webui/settings/settings_localized_strings_provider.cc
@@ -2078,7 +2078,7 @@ void AddSyncControlsStrings(content::Web
 void AddPeopleStrings(content::WebUIDataSource* html_source, Profile* profile) {
   static constexpr webui::LocalizedString kLocalizedStrings[] = {
       // Top level people strings:
-      {"peoplePageTitle", IDS_SETTINGS_PEOPLE},
+      {"peoplePageTitle", IDS_PROFILES_MENU_NAME},
       {"syncSettingsSavedToast", IDS_SETTINGS_SYNC_SETTINGS_SAVED_TOAST_LABEL},
       {"manageGoogleAccount", IDS_SETTINGS_MANAGE_GOOGLE_ACCOUNT},
       {"syncAndNonPersonalizedServices",
